---
title: "Lens transmission across anuran ecology"
author: "Katie Thomas"
date: "17 February 2020"
output:
  html_document:
    theme: cerulean
    toc: TRUE
    toc_float: TRUE
    code_fold: hide
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}

library(knitr)
library(kableExtra)
library(grid)
library(gtable)
library(ggtree)
library(ape)
library(plotly)
library(plyr)
library(geiger)
library(caper)
library(phytools)
library(AmphiNom)
library(cowplot)
library(tidyverse)
library(dplyr)

#set rmarkdown options
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, fig.height = 5, fig.width = 8, attr.output='style="max-height: 350px;"') 

```

# Data

## Lens transmission data

### This study

Lens data were collected from frozen and then thawed lenses in this study. The data span two orders (Anura and Caudata), 29 families, 62 genera, and 86 species. Most data are from post-metamorphic frogs, but we also have data on two tadpoles.

```{r lens-data}

#import species data for lens transmission
lens_raw <- data.frame(read.csv("../Data/species_data.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

#tidy data
lens_data <- lens_raw %>%
  #put genus and species in separate columns
  separate(species, c("genus", "species"), sep = " ", extra = "drop") %>%
  #add genus_species
  mutate(genus_species = as.factor(paste(genus, species, sep = "_"))) %>%
  mutate(dataset = "this_study") %>%
  mutate(source = "this_study") %>%
  mutate(scanned = "frozen lens") %>%
  select(order, family, genus_species, stage, dataset, source, diam_mm, approx_size, t450nm, t50, pUVA)

```

### Compiled from literature

Lens transmission data measured from fresh anuran lenses were gathered from previously published literature.  Note that for _Bombina orientalis_ the T50% was <300nm. To be able to include it here, I have set it at 300 so that it is numerical and can be included in models and visuals. These data span 14 families, 28 genera,  and 37 species (all post-metamorphic frogs). 

```{r lens-data-lit}

#import species data for lens transmission
lens_lit_raw <- data.frame(read.csv("../Data/published_data.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

#tidy data
lens_lit <- lens_lit_raw %>%
  mutate(genus_species = as.factor(species)) %>%
  mutate(t50 = as.numeric(replace(t50, t50=="<300", 300))) %>%
  mutate(t450nm = t450nm*100) %>%
  mutate(source = "literature") %>%
  mutate(stage = "unknown") %>%
  mutate(diam_mm = "unknown") %>%
  select(order, family, genus_species, stage, dataset, source, diam_mm, t450nm, t50, pUVA)
```

### Merged data

Data from our study and previous studies were merged for comparative analyses. Combined, they cover two orders, 32 families, 76 genera, and 116 species. There was overlap in data (our study and in published literature) for 8 species.  

```{r merge-data}

#merge new and published datasets 
lenses <-full_join(lens_data, lens_lit) 

#compile number of frogs sampled by species
samps <- lenses %>%
  arrange(genus_species) %>%
  arrange(family) %>%
  arrange(order) %>%
  select(order, family, genus_species, source, t50, pUVA)

#create table column names
names(samps) <- c("Order","Family", "Species", "Data source", "T50%", "%UVA") 

#generate scrolling table in RMarkdown
kable(samps[ , c("Order","Family", "Species", "Data source", "T50%", "%UVA")], 
      caption = "Species sampled and transmission summary data") %>%
  kable_styling(full_width = F) %>% 
  collapse_rows(columns = c(1,2), valign = "top") %>%
  scroll_box(height = "600px") 
```

## Ecological trait data

Ecological data for each species was imported and matched to the species in the lens transmission dataset. In the original trait database used for other projects, adult habitat was categorized as: fossorial, subfossorial, aquatic, semiaquatic, ground-dwelling, scansorial; here, we collapse it into binary states (scansorial or other) to test whether scansoriality is correlated with lens transmission. Likewise, in our original database, activity pattern was categorized as: diurnal, nocturnal, both; here, we make it binary (primarily diurnal or not) to test whether diurnal activity is associated with lens transmission. Finally, we include data for the presence or absence of sexual dichromatism in each species. 

```{r trait-data}
#import trait data
traits_raw <- data.frame(read.csv("../Data/lens_traits.csv", header=TRUE, na.strings=c("", "NA")))

#convert to binary traits and extract relevent columns
traits <- traits_raw %>%
  mutate(hab = recode_factor(Adult_habitat, "Scansorial" = "scansorial", "Fossorial" = "other", "Subfossorial" = "other", "Ground-dwelling" = "other", "Aquatic" = "other", "Semiaquatic" = "other")) %>%
  mutate(Activity_period = na_if(Activity_period, "Unknown")) %>%
  mutate(act = recode_factor(Activity_period, "Both" = "nondiurnal", "Diurnal" = "diurnal", "Nocturnal" = "nondiurnal")) %>%
  mutate(Sex_dichromatism = na_if(Sex_dichromatism, "Unknown")) %>%
  mutate (dich = recode_factor(Sex_dichromatism, "Absent" = "nondichromatic", "Present" = "dichromatic")) %>%
  select(genus_species, hab, act, dich)

#merge lens and trait data by species name
lens_traits <- left_join(lenses, traits, by="genus_species")
```

### Phylogeny

We used the amphibian phylogenetic hypothesis by Jetz and Pyron (2019). It's important to note that this tree includes branches supported by molecular data as well as branches that are grafted on based exclusively on taxonomy, and has many polytomies. Because we need a dichotomous tree for comparative analyses, we randomly resolved polytomies using the multi2di function in ape v.5.4.1. 

```{r pyrontree, results = "hide"}

#Import phylogeny from Jetz and Pyron 2019
tree_orig <- read.tree(file = "../Data/amph_shl_new_Consensus_7238.tre") #reads tree

#Test whether tree is rooted
is.rooted(tree_orig)
```

We matched up the species names in the tree to the species names in our dataset by converting both species list to Frost's taxonomy from Amphibian Species of the World (ASW). Anuran taxonomy changes frequently and there are likely some species  that are named differently in the tree vs. in our datasets. To match them up, we use the AmphiNom package by Liedtke (2019) to convert known synonyms to the taxonomy represented in Amphibian Speicies of the World (ASW). This addressed the bulk of the taxa; the remaining we examined and matched manually. 

```{r name-match, cache = TRUE, results="hide"}

#update AmphiNom package with most current ASW taxonomy
##note: this is time consuming but should be redone periodically for most updated results.
### last update: 10 December 2020
#getTaxonomy() 

### Dataset names ------

#Screen names in dataset to see how well they match up to ASW and what can be 'updated' seamlessly
datanames.asw <- aswSync(lens_traits$genus_species)
synonymReport(datanames.asw)

#pull out all direct matches or updated ASW taxonomy
data.update <- datanames.asw %>%
  filter(status == "up_to_date" | status =="updated") %>%
  mutate(genus_species = query) %>%
  select(genus_species, ASW_names)

#add column to lens data with updated ASW names
lens_names <- distinct(left_join(lens_traits, data.update, by = "genus_species"))

#check how many species aren't being matched to ASW names
missingASW <- lens_names %>%
  filter(is.na(ASW_names)) %>%
  select(genus_species, ASW_names)

#add ASW manual entries after checking species
missingASW$ASW_names[missingASW$genus_species=="Cornufer_guentheri"]<-"Cornufer_guentheri" #ambuiguous (mult. matches)
missingASW$ASW_names[missingASW$genus_species=="Lissotriton_vulgaris"]<-"Lissotriton_vulgaris" #ambuiguous (mult. matches)
missingASW$ASW_names[missingASW$genus_species=="Cynops_pyrrhogaster"]<-"Cynops_pyrrhogaster" #ambiguous (mult. matches)

#merge these updated names with original data update
data.update2 <- full_join(data.update, missingASW)

#remerge with pupil data
lens_names2 <- distinct(left_join(lens_traits, data.update2, by = "genus_species"))

### Phylogeny names -----

#Pull out tip labels from tree
tips <- as.vector(tree_orig$tip.label)

#Make a dataframe with tip vector as a column called "phylo_names"
phylo_names <- as.data.frame(tips, optional = TRUE, stringsAsFactors = FALSE)
colnames(phylo_names) <- "species_phylo"

#Screen names in phylogeny tips to see how well they match up to ASW and what can be 'updated' seamlessly
phylotips.asw <- aswSync(phylo_names$species_phylo)

#view summary
synonymReport(phylotips.asw)

#pull out all direct matches for updated ASW taxonomy
tip.update <- phylotips.asw %>%
  filter(status == "up_to_date" | status =="updated") %>%
  mutate(species_phylo = query) %>%
  select(species_phylo, ASW_names) 

#add column to phylotips with updated ASW names
phylo_names2 <- left_join(phylo_names, tip.update, by = "species_phylo")

### Find species missing from tips and dataset ----

#check how many species in ASW revised dataset are not matching new tree ASW labels
length(which(!lens_names2$ASW_names %in% phylo_names2$ASW_names))

#list species that need manually checked in tree
missing <- setdiff(lens_names2$ASW_names, phylo_names2$ASW_names)
missing <- as.data.frame(missing, optional = TRUE, stringsAsFactors = FALSE)
colnames(missing) <- "ASW_names" #rename col with ASW names
missing$species_phylo <- NA #create empty col for phylo matches

#add ASW phylo matches manually 

#Add Lissotriton vulgaris (was ambiguous match)
missing$species_phylo[missing$ASW_names=="Lissotriton_vulgaris"]<-"Lissotriton_vulgaris" 

#Add Cynops pyrrhogaster (was ambiguous match)
missing$species_phylo[missing$ASW_names=="Cynops_pyrrhogaster"]<-"Cynops_pyrrhogaster" 

#Add Lithobates pipiens
missing$species_phylo[missing$ASW_names=="Lithobates_pipiens"]<-"Rana_pipiens" 

#Boana diabolica is missing from phylogeny and will have to be grafted in as a sister taxon to Boana geographica (according to Caminer and Ron 2020)) later
# Adding it here as placeholder now. 
missing$species_phylo[missing$ASW_names=="Boana_diabolica"]<-"Boana_diabolica" #ambiguous (mult. matches)


#add manual matches to phylogeny ASW matches
phylo_names3 <- phylo_names2 %>%
  drop_na(ASW_names)
  
phylo_names4 <- full_join(phylo_names3, missing)

#check how many species in ASW revised dataset are not matching new tree ASW labels
missing2 <- setdiff(lens_names2$ASW_names, phylo_names4$ASW_names)

#check for duplicate taxa
n_occur <- data.frame(table(lens_names2$genus_species))
#View(n_occur)

### Match data species names with phylogeny tip labels ###

#remove duplicate ASW matches from phylogeny labels
phylo_names5 <- phylo_names4 %>%
  distinct(ASW_names, .keep_all = TRUE) #remove any duplicates

#use a left join in dplyr to match phylo tips to dataset species names
phylo_data <- lens_names2 %>%
  left_join(phylo_names5, by = "ASW_names") %>%
  drop_na(species_phylo) %>% #drop species not in phylogeny
  mutate_if(is.factor, as.character) %>%
  distinct(ASW_names, .keep_all = TRUE) #remove any duplicates

### Pruning a phylogeny to match the species in your dataset ###

#make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(tree_orig$tip.label, phylo_data$species_phylo)

#drop unwanted tips from phylogeny
tree.pruned <- drop.tip(phy = tree_orig, tip = drops) 

#see which tips you've kept in your phylogeny
tree.pruned$tip.label

#rename tips to ASW names
tree.pruned$tip.label <- phylo_data[["ASW_names"]][match(tree.pruned$tip.label, phylo_data[["species_phylo"]])]

#force species_phylo to dataframe
phylo_data <- as.data.frame(phylo_data)
rownames(phylo_data) <- phylo_data$ASW_names

#check that phylogeny tips and data match exactly (if they match will return "OK")
name.check(tree.pruned, phylo_data)

#check and reorder tree
is.rooted(tree.pruned) #tests whether tree is rooted (want it to return "TRUE") 

#Test whether tree is ultrametric
is.ultrametric(tree.pruned)

#Force tree to be ultrametric (it should be)
tree_edit <- force.ultrametric(tree.pruned)

#Boana diabolica is missing from phylogeny and will be grafted in as a polytomy based on the genus
tree_add <- add.species.to.genus(tree_edit, "Boana_diabolica", where = "root")

#test whether tree is dichotomous (no polytomies)
is.binary.tree(tree_add) 

#make tree dichotomous
tree_add2 <- multi2di(tree_add)

#ladderize 
lens.tree <- ladderize(tree_add2) #ladderizes tree
```


```{r plot-phylo, fig.height = 20, fig.width = 8, fig.cap="Phylogeny from Jetz and Pyron (2019), pruned to the taxa in our study and re-named so that species match taxonomy on ASW."}
#plot phylogeny
plot.phylo(lens.tree, #phylogeny to plot
           type = "phylogram", #how to visualize phylogeny
           show.tip.label = TRUE, #whether to show tip labels/species names
           cex = 1.0, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE,
           edge.width = 1.5, #thickness of phylogeny branches
           label.offset = 3) #how far away from tips to put labels
```

### Analysis subsets

We subset three groups for visualization and analyses by life stage and dataset: adults (includes our adult data and all published literature data), tadpoles (our data only), a pair of _Hyllodes phyllodes_, one juvenile and one adult that showed different strengths of UV filtering, likely due to the size of the frog/lens resulting in differences in path length of the light, and a tadpole/adult pair of _Alytes muletensis_ (our only tad/adult match in the dataset).

```{r subsets}

#make dataframe to match ASW names/tip labels to full lens dataframe
dat.names <- phylo_data %>%
  select(genus_species, ASW_names)

#match phylogeny tip labels (ASW names) with lens dataframe
lenses.phy <- left_join(lens_traits, dat.names, by = "genus_species")
  
#adults only
lens_adults <- lenses.phy %>%
  filter(stage %in% c("adult", "unknown"))

#adults-this study
lens_adults.new <- lens_adults %>% 
  filter(source == "this_study")

#adults-literature
lens_adults.pub <- lens_adults %>% 
  filter(source == "literature")

#tads only
lens_tads <- lenses.phy %>%
  filter(stage == "tadpole")

#hyllodes set
lens_hyl <- lenses.phy %>%
  filter(genus_species == "Hylodes_phyllodes")

#tad/adult match
lens_dev <- lenses.phy %>%
  filter(genus_species == "Alytes_muletensis")
```

# Lens transmission vs. lens size

Here, we run PGLS regressions of t50 vs. log10 lens diameter and of %UVA vs. log10 lens diameter to test for relationships between transmission and path length. 

## Dataset: all adult species means

Here, we included only adult data from our study (we excluded published adult data because lens sizes from specimens used for transmission measurements were not reported). We include all species, whether or not lens transmission spectra appear to indicate the presence of a pigment. This was to test the null hypothesis that variation in transmission is due to different spectral properties in lenses across species. If we find a significant, highly correlated relationship between transmission and pathlength, it could indicate that all lenses have the same spectral properties, and differences in transmission may be driven only by differences in path length across species with different eye sizes. 

### t50 vs. path length

```{r, results = "hide"}

#PGLS models for t50 or %UVA vs. lens diameter among species

#### prep data ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(lens.tree$tip.label, as.character(lens_adults.new$ASW_names))

#Drop unwanted tips from phylogeny
adults.new.tree <- drop.tip(phy = lens.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(lens_adults.new) <- lens_adults.new$ASW_names

#check that names match in dataframe and tree
name.check(phy = adults.new.tree, data = lens_adults.new, data.names = lens_adults.new$ASW_names)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
lenssize.comp <- comparative.data(phy = adults.new.tree, data = lens_adults.new, 
                              names.col = ASW_names, vcv = TRUE, 
                              na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
lenssize.comp$dropped$tips #phylogeny
lenssize.comp$dropped$unmatched.rows #dataset

#### PGLS model for t50 vs. lens size ####

#model
pgls_size.t50 <- pgls(t50 ~ log10(approx_size), 
                  data = lenssize.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)
```

After fitting the model, we can examine the diagnostic plots to check model assumptions, then look at the parameter estimates and the likelihood profile of lambda. 

```{r}
###check model assumptions ###

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_size.t50, main = "PGLS model for t50 vs. log lens size")
par(mfrow = c(1, 1))

### model outputs ###

#print model output 
summary(pgls_size.t50)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.size.t50 <- pgls.profile(pgls_size.t50, "lambda")
plot(lambda.size.t50)

```

Finally, we can plot our data and the model fit. Note that the linear relationship between t50 and log lens diameter was not significant, indicating that observed variation in transmission among species is not driven solely by differences in path length. 

```{r, fig.height = 4, fig.width = 6}

#plot t50 vs. log lens size with fit
plot_t50size <- ggplot(lens_adults.new, aes(x = approx_size, y = t50, text = genus_species)) + 
  geom_point(size = 2, alpha = 0.8) + 
  scale_x_log10("Lens diameter (mm)") + 
  ylab("t50") + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  geom_abline(slope = coef(pgls_size.t50)[[2]], intercept = coef(pgls_size.t50)[[1]], linetype = "dashed")

#interactive plot
ggplotly(plot_t50size)

```

## %UVA vs. path length

```{r, results = "hide"}

#### PGLS model for %UVA vs. lens size ####

#model
pgls_size.puva <- pgls(pUVA ~ log10(approx_size), 
                  data = lenssize.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)

```

After fitting the model, we can examine the diagnostic plots to check model assumptions, then look at the parameter estimates and the likelihood profile of lambda.

```{r}
###check model assumptions ###

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_size.puva, main = "PGLS model for pUVA vs. log lens size")
par(mfrow = c(1, 1))

### model outputs ###

#print model output 
summary(pgls_size.puva)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.size.uva <- pgls.profile(pgls_size.puva, "lambda")
plot(lambda.size.uva)

```

Finally, we can plot our data and the model fit. Note that the linear relationship between %UVA and log lens diameter was not significant, indicating that observed variation in transmission among species is not driven solely by differences in path length.

```{r, fig.height = 4, fig.width = 6}
#plot %UVA vs. log lens size with fit
plot_uvasize <- ggplot(lens_adults.new, aes(x = approx_size, y = pUVA, text = genus_species)) + 
  geom_point(size = 2, alpha = 0.8) + 
  scale_x_log10("Lens diameter (mm)") + 
  ylab("%UVA") + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  geom_abline(slope = coef(pgls_size.puva)[[2]], intercept = coef(pgls_size.puva)[[1]], linetype = "dashed")

#interactive plot
ggplotly(plot_uvasize)

```

## Dataset: adult species means with unpigmented lenses 

We only expect path length to show a strong relationship with UV transmission in unpigmented lenses, so we may want to look separately at just unpigmented ones. In our results, we considered a t50 ≤ 350nm to be an unpigmented lens (n = 27).

Our lenses ranged in t50 from 317 to 423.  We can look at that plotted onto a histogram of all our t50 values to see where that falls in our data. 

```{r}

#Histogram of T50 values in our dataset
  plot_t50hist <-ggplot(lens_adults.new, aes(x = t50)) + 
  geom_histogram(color = "black", fill = "gray", binwidth = 5) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  xlab("t50") +
  geom_vline(aes(xintercept=350), col = "red", linetype = "dashed")
  

plot_t50hist
```

### t50 vs. path length (unpigmented)

```{r, results = "hide"}

#add pigmentation category to dataframe
lens_adults.new <- lens_adults.new %>%
  mutate(pigment = ifelse(t50 <= 350, "unpigmented", "pigmented"))

#subset unpigmented species
lens_unpig <- lens_adults.new %>%
  filter(pigment=="unpigmented")

#PGLS models for t50 or %UVA vs. lens diameter among species

#### prep data ####

#Make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(lens.tree$tip.label, as.character(lens_unpig$ASW_names))

#Drop unwanted tips from phylogeny
unpig.tree <- drop.tip(phy = lens.tree, tip = drops) 

#Make row names of the species the phylogeny tip labels
rownames(lens_unpig) <- lens_unpig$ASW_names

#check that names match in dataframe and tree
name.check(phy = unpig.tree, data = lens_unpig, data.names = lens_unpig$ASW_names)

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
unpig.comp <- comparative.data(phy = unpig.tree, data = lens_unpig, 
                              names.col = ASW_names, vcv = TRUE, 
                              na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
unpig.comp$dropped$tips #phylogeny
unpig.comp$dropped$unmatched.rows #dataset

#### PGLS model for t50 vs. lens size ####

#model
pgls_unpig.t50 <- pgls(t50 ~ log10(approx_size), 
                  data = unpig.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)

```

After fitting the model, we can examine the diagnostic plots to check model assumptions, then look at the parameter estimates and the likelihood profile of lambda. 

```{r}
###check model assumptions ###

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_unpig.t50, main = "PGLS model for t50 vs. log lens size in unpigmented lenses")
par(mfrow = c(1, 1))

### model outputs ###

#print model output 
summary(pgls_unpig.t50)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.unpig.t50 <- pgls.profile(pgls_unpig.t50, "lambda")
plot(lambda.unpig.t50)

```

Finally, we can plot our data and the model fit. Note that the linear relationship between t50 and log lens diameter was not significant, indicating that lens transmission among species with unpigmented lenses is not correlated with path length.

```{r, fig.height = 4, fig.width = 6}
#plot t50 vs. log lens size with fit
plot_t50unpig <- ggplot(lens_unpig, aes(x = approx_size, y = t50, text = genus_species)) + 
  geom_point(size = 2, alpha = 0.8) + 
  scale_x_log10("Lens diameter (mm)") + 
  ylab("t50") + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  geom_abline(slope = coef(pgls_unpig.t50)[[2]], intercept = coef(pgls_unpig.t50)[[1]], linetype = "dashed")

#interactive plot
ggplotly(plot_t50unpig)
```

### %UVA vs. path length (unpigmented)

```{r}
#### PGLS model for %UVA vs. lens size ####

#model
pgls_unpig.puva <- pgls(pUVA ~ log10(approx_size), 
                  data = unpig.comp, 
                  lambda = "ML", #uses Maximum Liklihood estimate of lambda
                  param.CI = 0.95)
```

After fitting the model, we can examine the diagnostic plots to check model assumptions, then look at the parameter estimates and the likelihood profile of lambda. 

```{r}
###check model assumptions ###

#diagnostic plots
par(mar = c(4,4,2,2))
par(mfrow = c(2, 2))
plot(pgls_unpig.puva, main = "PGLS model for pUVA vs. log lens size in unpigmented lenses")
par(mfrow = c(1, 1))

### model outputs ###

#print model output 
summary(pgls_unpig.puva)

#Likelihood plot for Pagel's lambda from the PGLS model of eye diameter vs. the cuberoot of mass. Solid red line indicates estimate for lambda and broken red lines indcaite the 95% confidence interval
lambda.unpig.uva <- pgls.profile(pgls_unpig.puva, "lambda")
plot(lambda.unpig.uva)
```
Finally, we can plot our data and the model fit. Note that the linear relationship between t50 and log lens diameter was not significant, indicating that lens transmission among species with unpigmented lenses is not correlated with path length.

```{r, fig.height = 4, fig.width = 6}
#plot %UVA vs. log lens size with fit
plot_uva.unpig <- ggplot(lens_unpig, aes(x = approx_size, y = pUVA, text = genus_species)) + 
  geom_point(size = 2, alpha = 0.8) + 
  scale_x_log10("Lens diameter (mm)") + 
  ylab("%UVA") + 
  theme_bw() +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  geom_abline(slope = coef(pgls_unpig.puva)[[2]], intercept = coef(pgls_unpig.puva)[[1]], linetype = "dashed")

#interactive plot
ggplotly(plot_uva.unpig)

```

The lack of relationship between transmission and lens size across species is good evidence that differences we observe across species are due to differences in the spectral properties of the lens, even among species with highly transparent lenses. It also justifies why we do not account for lens diameter in our models of lens transmission vs. ecology (which is impossible, as the published data we include did not have measurements for lens diameter).


# Visualize ecological data

## Plot adult sampling onto phylogeny

```{r, results = "hide"}

# prep data -----

#create dataframe of adults with this paper and lit data in two separate columns

#our data
new_cols <- lens_adults.new %>%
  mutate(t450nm_new = t450nm) %>%
  mutate(t50_new = t50) %>%
  mutate(pUVA_new = pUVA) %>%
  select(ASW_names, t450nm_new, t50_new, pUVA_new, hab, act, dich)

#published data
lit_cols <- lens_adults.pub %>%
  mutate(t450nm_lit = t450nm) %>%
  mutate(t50_lit = t50) %>%
  mutate(pUVA_lit = pUVA) %>%
  select(ASW_names, t450nm_lit, t50_lit, pUVA_lit, hab, act, dich)

#merge
adults_sep <- full_join(new_cols, lit_cols, by = c("ASW_names", "hab", "act", "dich"))

#prune tree to adult data -----

#make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(lens.tree$tip.label, adults_sep$ASW_names)

#drop unwanted tips from phylogeny
tree.adults <- drop.tip(phy = lens.tree, tip = drops) 

#check that phylogeny tips and data match exactly (if they match will return "OK")
name.check(tree.adults, phylo_data)

#resort dataset to the order of tree tip labels
rownames(adults_sep) <- adults_sep$ASW_names
adults_sep <- adults_sep[tree.adults$tip.label, ] 

```

```{r, fig.height=12, fig.width=7, fig.cap="Phylogeny showing adult lens transmission and ecological traits. Non-italicized values show our t50 (unbolded) and %UVA (bolded) measurements for adult species means, while italicized values show species means calculated from published literature."} 

#plot phylogeny with sampling tip labels ----

#phylogeny
plot.phylo(tree.adults, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.6, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE, 
           edge.width = 2,
           x.lim = 650,
           label.offset = 2) 

#add tip labels for our data (t50)
tiplabels(adults_sep$t50_new,
          cex = 0.6,
          frame = "none",
          bg = "white",
          offset = 195) #size of labels

#add tip labels for published data (t50) 
tiplabels(adults_sep$t50_lit,
          cex = 0.6,
          font = 3,
          frame = "none",
          bg = "white",
          offset = 215) #size of labels

#add tip labels for our data (%UVA)
tiplabels(round(adults_sep$pUVA_new, digits = 0),
          cex = 0.6,
          font = 2,
          frame = "none",
          bg = "white",
          offset = 240) #size of labels

#add tip labels for published data (%UVA)
tiplabels(round(adults_sep$pUVA_lit,digits = 0),
          cex = 0.6,
          font = 4, 
          frame = "none",
          bg = "white",
          offset = 255) #size of labels

#add tip labels for scansorial/not
tiplabels(col = c("#009E73","grey95")[adults_sep$hab], #sets color habitat
          pch = 17,
          cex = 0.8,
          offset = 150) 

#add tip labels for diurnal/not
tiplabels(col = c("grey95","#FFB122")[adults_sep$act], #sets color habitat
          pch = 19,
          cex = 0.8,
          offset = 160) 

#add tip labels for sexually dimorphic/not
tiplabels(col = c("grey95","#03a1fc")[adults_sep$dich], #sets color habitat
          pch = 15,
          cex = 0.8,
          offset = 170) 

#add legend for ecology traits
legend(x = 20, y = 25, legend = c("Scansorial", "Diurnal", "Sexually dichromatic"), 
       col = c("#009E73", "#FFB122", "#03a1fc"),
       pch = c(17,19,15), #shape of labels
       cex = 0.6, 
       box.lty = 0, 
       title = "Ecology traits", 
       title.adj = 0)

```

```{r, results = "hide"}
#export pdf

#pdf
pdf("../Figures/phylogeny_summary.pdf", width = 7, height = 12)

#phylogeny
plot.phylo(tree.adults, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.6, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE, 
           edge.width = 2,
           x.lim = 650,
           label.offset = 2) 

#add tip labels for our data (t50)
tiplabels(adults_sep$t50_new,
          cex = 0.6,
          frame = "none",
          bg = "white",
          offset = 195) #size of labels

#add tip labels for published data (t50) 
tiplabels(adults_sep$t50_lit,
          cex = 0.6,
          font = 3,
          frame = "none",
          bg = "white",
          offset = 215) #size of labels

#add tip labels for our data (%UVA)
tiplabels(round(adults_sep$pUVA_new, digits = 0),
          cex = 0.6,
          font = 2,
          frame = "none",
          bg = "white",
          offset = 240) #size of labels

#add tip labels for published data (%UVA)
tiplabels(round(adults_sep$pUVA_lit,digits = 0),
          cex = 0.6,
          font = 4, 
          frame = "none",
          bg = "white",
          offset = 255) #size of labels

#add tip labels for scansorial/not
tiplabels(col = c("#009E73","grey95")[adults_sep$hab], #sets color habitat
          pch = 17,
          cex = 0.8,
          offset = 150) 

#add tip labels for diurnal/not
tiplabels(col = c("grey95","#FFB122")[adults_sep$act], #sets color habitat
          pch = 19,
          cex = 0.8,
          offset = 160) 

#add tip labels for sexually dimorphic/not
tiplabels(col = c("grey95","#03a1fc")[adults_sep$dich], #sets color habitat
          pch = 15,
          cex = 0.8,
          offset = 170) 

#add legend for ecology traits
legend(x = 20, y = 25, legend = c("Scansorial", "Diurnal", "Sexually dichromatic"), 
       col = c("#009E73", "#FFB122", "#03a1fc"),
       pch = c(17,19,15), #shape of labels
       cex = 0.6, 
       box.lty = 0, 
       title = "Ecology traits", 
       title.adj = 0)

dev.off()

```

## Plots of t50 and %UVA values across ecological states

Here we plot adult lens transmission metrics (t50 and %UVA) across ecological categories to visualize differences (without accounting for phylogeny)

```{r cols, results = "hide"}

# create vectors of colors for habitat
col_hab <- c("scansorial" = "#009E73",
              "other"  = "#BDC3BE")

#create vector of colors for activity period
col_act <- c("nondiurnal" = "#BDC3BE",
             "diurnal" = "#FFB122")

#create vector of colors for activity period
col_dich <- c("nondichromatic" = "#BDC3BE",
              "dichromatic" = "#03a1fc")

#create shape vector for source of data
sh_source <- c("literature" = 1, 
               "this_study" = 19)
```


```{r boxplots-eco}

# Scansoriality ------

# boxplot of T50 across habs
boxplot_habs.t50 <- ggplot(data = lens_adults, 
                       aes(x = hab, y = t50)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = hab, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab,
                     name = "Habitat",
                     breaks = c("scansorial", "other"),
                     labels = c("scansorial", "non-scansorial")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("") +
  ylab("t50 (nm)")

# boxplot of %UVA across habs
boxplot_habs.uva <- ggplot(data = lens_adults, 
                       aes(x = hab, y = pUVA)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = hab, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab,
                     name = "Habitat",
                     breaks = c("scansorial", "other"),
                     labels = c("scansorial", "non-scansorial")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("Scansoriality") +
  ylab("%UVA") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_habs.t50, boxplot_habs.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids

# Diurnality ------

# boxplot of T50 across acts
boxplot_act.t50 <- ggplot(data = lens_adults, 
                       aes(x = act, y = t50)) + 
   geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = act, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act,
                     name = "Activity",
                     breaks = c("diurnal", "nondiurnal"),
                     labels = c("diurnal", "non-diurnal")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("Diurnality") +
  ylab("t50 (nm)") +
  labs(fill = "Data source")

# boxplot of %UVA across habs
boxplot_act.uva <- ggplot(data = lens_adults, 
                       aes(x = act, y = pUVA)) + 
   geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = act, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act,
                     name = "Activity",
                     breaks = c("diurnal", "nondiurnal"),
                     labels = c("diurnal", "non-diurnal")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("Diurnality") +
  ylab("%UVA") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_act.t50, boxplot_act.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids


# Sexual dichromatism ------

# boxplot of T50 across habs
boxplot_dich.t50 <- ggplot(data = lens_adults, aes(x = dich, y = t50)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) +
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) +
 geom_jitter(aes(color = dich, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich,
                     name = "Sexual dichromatism",
                     breaks = c("dichromatic", "nondichromatic"),
                     labels = c("dichromatic", "non-dichromatic")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("Sexual dichromatism") +
  ylab("t50 (nm)") +
  labs(fill = "Data source")

# boxplot of %UVA across habs
boxplot_dich.uva <- ggplot(data = lens_adults, 
                       aes(x = dich, y = pUVA)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.9) +
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) +
 geom_jitter(aes(color = dich, shape = source), size = 1.3, alpha = 0.7, position = position_jitter(0.15)) +
  scale_color_manual(values = col_dich,
                     name = "Sexual dichromatism",
                     breaks = c("dichromatic", "nondichromatic"),
                     labels = c("dichromatic", "non-dichromatic")) +
  scale_shape_manual(values = sh_source,
                     name = "Data",
                     breaks = c("this_study", "literature"),
                     labels = c("this study", "literature"),
                     guide = guide_legend(override.aes = list(color = "black"), fill = "black")) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"), legend.key = element_rect(fill = NA)) + #controls background
  xlab("Sexual dichromatism") +
  ylab("%UVA") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_dich.t50, boxplot_dich.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids

```


```{r boxplots-eco-export, results = "hide"}

#Boxplot figure

fig.a <- boxplot_habs.t50 + 
  xlab("")+ 
  ylab("t50") +
  theme(legend.position = "none")

fig.b <- boxplot_habs.uva + 
  xlab("")+ 
  ylab("%UVA") +
  theme(legend.position = "none")

fig.c <- boxplot_act.t50 + 
  xlab("")+ 
  ylab("t50") +
  theme(legend.position = "none")

fig.d <- boxplot_act.uva + 
  xlab("")+ 
  ylab("%UVA") +
  theme(legend.position = "none")

fig.e <- boxplot_dich.t50 + 
  xlab("")+ 
  ylab("t50") +
  theme(legend.position = "none")

fig.f <- boxplot_dich.uva + 
  xlab("")+ 
  ylab("%UVA") +
  theme(legend.position = "none")

# boxplot of T50 across habs
pdf("../Figures/boxplots.pdf", width = 7, height = 10)
plot_grid(fig.a, fig.b, fig.c, fig.d, fig.e, fig.f,
          align = 'vh', 
          labels = c("A", "B", "C", "D", "E", "F"),
          hjust = -1, #adjustment for panel labels
          ncol = 2)
dev.off()

```


## Plots of t50 and %UVA values onto phylogeny

Here and in subsequent analyses, we must have one transmission value assigned to each species. We include data from both our study and from previously published literature. However, for the 8 species with data from both sources, we use our data in these visualizations and in analyses.

```{r subset-adultphy, results = "hide"}

#For species with more than one data point (measured by us and in lit), choose our data
adults_merged <- adults_sep %>%
  mutate(t50_merge = ifelse(is.na(adults_sep$t50_new), t50_lit, t50_new)) %>%
  mutate(pUVA_merge = ifelse(is.na(adults_sep$pUVA_new), pUVA_lit, pUVA_new))
```

```{r prep-ggtree, results = "hide"}

#subset data
adults_plot <- adults_merged %>%
  mutate(tip = ASW_names) %>%
  #put genus and species in separate columns
  separate(ASW_names, c("genus", "species"), sep = "_", extra = "drop") %>%
  #add tip labels with regular text
  mutate(labels = as.factor(paste(genus, species, sep = " "))) %>%
  select(tip, labels, t50_merge, pUVA_merge, hab, act, dich)

# set row names in dataset to match the tip labels in the tree
row.names(adults_plot) <- adults_plot$tip

#check that phylogeny and data match exactly
name.check(tree.adults, adults_plot)

#resort trait dataset to the order of tree tip labels
adults_plot <- adults_plot[tree.adults$tip.label, ] 

#labels
labs <- adults_plot %>% select(tip, labels)
```

#### plot data

First we visualize the distribution of all our t50 and %UVA data across the phylogeny. 

```{r ggtree-data, fig.width = 10, fig.height=15}

# Make the phylogeny plot
p <- ggtree(tree.adults) %<+% labs + 
     geom_tiplab(cex = 3, aes(label = labels), fontface = 3) + 
   xlim_tree(450) + 
   coord_cartesian(clip = 'off') 

# Make a second plot next to the phylogeny (dot plot of t50 values)
p2 <- facet_plot(p, panel="t50 (nm)", data=adults_plot, geom=geom_point, aes(x=t50_merge), color = "#BDC3BE",  shape = 19, cex = 2.5)

# Make a third plot next to the first two (bar plot of %UVA)
p3 <- facet_plot(p2, panel='%UVA', data=adults_plot, geom=geom_segment, aes(x=0, xend=pUVA_merge, y=y, yend=y), color = "#BDC3BE", size=3) +
  theme_tree2(legend.position=c(.07, .93)) #add scale bars and move legends

#control widths of panels
 gt = ggplot_gtable(ggplot_build(p3))
# gtable_show_layout(gt) # will show you the layout - very handy function
# gt # see plot layout in table format
# gt$layout$l[grep('tree', gt$layout$name)] # you want to find the column specific to panel-2
 gt$widths[7] = 0.5*gt$widths[7] # reduce column 7
 gt$widths[9] = 0.5*gt$widths[9] 

#print figure
p3
```

```{r results = "hide"}
#export as pdf
pdf("../Figures/ggtree-data.pdf", width = 10, height = 15)
grid::grid.draw(gt) # plot with grid draw
dev.off()
```


#### plot data + scansoriality

Then we visualize our data colored by habitat. 

```{r ggtree-scansorial, fig.width = 10, fig.height=15}

# Make the phylogeny plot
p <- ggtree(tree.adults) %<+% labs + 
     geom_tiplab(cex = 3, aes(label = labels), fontface = 3) + 
   xlim_tree(450) + 
   coord_cartesian(clip = 'off') 

# Make a second plot next to the phylogeny (dot plot of t50 values)
p2 <- facet_plot(p, panel="t50 (nm)", data=adults_plot, geom=geom_point, aes(x=t50_merge, color = hab), shape = 19, cex = 2.5) +
  scale_color_manual(values = col_hab,
                     name = "Habitat",
                     breaks = c("scansorial", "other"),
                     labels = c("scansorial", "non-scansorial"))

# Make a third plot next to the first two (bar plot of %UVA)
p3 <- facet_plot(p2, panel='%UVA', data=adults_plot, geom=geom_segment, aes(x=0, xend=pUVA_merge, y=y, yend=y, color = hab), size=3) +
  theme_tree2(legend.position=c(.07, .93)) #add scale bars and move legends

#control widths of panels
 gt = ggplot_gtable(ggplot_build(p3))
# gtable_show_layout(gt) # will show you the layout - very handy function
# gt # see plot layout in table format
# gt$layout$l[grep('tree', gt$layout$name)] # you want to find the column specific to panel-2
 gt$widths[7] = 0.5*gt$widths[7] # reduce column 7
 gt$widths[9] = 0.5*gt$widths[9] 

#print figure
p3
```

```{r results = "hide"}
#export as pdf
pdf("../Figures/ggtree-scans.pdf", width = 10, height = 15)
grid::grid.draw(gt) # plot with grid draw
dev.off()
```

#### plot data + diurnality

Then we visualize our data colored by activity pattern. 

```{r ggtree-diurnal, fig.width = 10, fig.height=15}

# Make the phylogeny plot
p <- ggtree(tree.adults) %<+% labs + 
     geom_tiplab(cex = 3, aes(label = labels), fontface = 3) + 
   xlim_tree(450) + 
   coord_cartesian(clip = 'off') 

# Make a second plot next to the phylogeny (dot plot of t50 values)
p2 <- facet_plot(p, panel="t50 (nm)", data=adults_plot, geom=geom_point, aes(x=t50_merge, color = act), shape = 19, cex = 2.5) +
  scale_color_manual(values = col_act,
                     name = "Activity period",
                     breaks = c("diurnal", "nondiurnal"),
                     labels = c("diurnal", "non-diurnal"))


# Make a third plot next to the first two (bar plot of %UVA)
p3 <- facet_plot(p2, panel='%UVA', data=adults_plot, geom=geom_segment, aes(x=0, xend=pUVA_merge, y=y, yend=y, color = act), size=3) +
  theme_tree2(legend.position=c(.07, .93)) #add scale bars and move legends

#control widths of panels
 gt = ggplot_gtable(ggplot_build(p3))
# gtable_show_layout(gt) # will show you the layout - very handy function
# gt # see plot layout in table format
# gt$layout$l[grep('tree', gt$layout$name)] # you want to find the column specific to panel-2
 gt$widths[7] = 0.5*gt$widths[7] # reduce column 7
 gt$widths[9] = 0.5*gt$widths[9] 

#print figure
p3
```

```{r results = "hide"}
#export as pdf
pdf("../Figures/ggtree-act.pdf", width = 10, height = 15)
grid::grid.draw(gt) # plot with grid draw
dev.off()
```


#### plot data + sexual dichromatism

Then we visualize our data colored by the presence of sexual dichromatism. 

```{r ggtree-dichromatism, fig.width = 10, fig.height=15}

# Make the phylogeny plot
p <- ggtree(tree.adults) %<+% labs + 
     geom_tiplab(cex = 3, aes(label = labels), fontface = 3) + 
   xlim_tree(450) + 
   coord_cartesian(clip = 'off') 

# Make a second plot next to the phylogeny (dot plot of t50 values)
p2 <- facet_plot(p, panel="t50 (nm)", data=adults_plot, geom=geom_point, aes(x=t50_merge, color = dich), shape = 19, cex = 2.5) +
  scale_color_manual(values = col_dich,
                     name = "Sexual dichromatism",
                     breaks = c("dichromatic", "nondichromatic"),
                     labels = c("dichromatic", "non-dichromatic"))


# Make a third plot next to the first two (bar plot of %UVA)
p3 <- facet_plot(p2, panel='%UVA', data=adults_plot, geom=geom_segment, aes(x=0, xend=pUVA_merge, y=y, yend=y, color = dich), size=3) +
  theme_tree2(legend.position=c(.07, .93)) #add scale bars and move legends

#control widths of panels
 gt = ggplot_gtable(ggplot_build(p3))
# gtable_show_layout(gt) # will show you the layout - very handy function
# gt # see plot layout in table format
# gt$layout$l[grep('tree', gt$layout$name)] # you want to find the column specific to panel-2
 gt$widths[7] = 0.5*gt$widths[7] # reduce column 7
 gt$widths[9] = 0.5*gt$widths[9] 

 #print figure
p3
```

```{r results = "hide"}
#export as pdf
pdf("../Figures/ggtree-dich.pdf", width = 10, height = 15)
grid::grid.draw(gt) # plot with grid draw
dev.off()
```

#### make paper figure

Then we visualize our data colored by ecological states that were significant in our phylogenetic ANOVAs (below). Here, we make a new column in our dataframe indicating the presence of diurnality, scansoriality, both (n = 1), and neither. This will show the distribution of both significant ecological effects, as well as show that there is only one species that is both diurnal and scansorial, indicating that these are separate effects. 

```{r, results = "hide"}
#make new category with scansorial, diurnal, both, or neither
lens_adults.new <- lens_adults.new %>%
  mutate(eco_cat = case_when(hab=="scansorial" & act=="diurnal" ~ "both",
                             hab == "scansorial" ~ "scansorial",
                             act == "diurnal" ~ "diurnal", 
                             hab=="other" & act=="nondiurnal" ~ "neither")) %>%
  mutate(eco_binary = recode_factor(eco_cat, "scansorial" = "SD", "diurnal" = "SD", "both" = "SD", "neither" = "not"))

#subset data
adults_plot.combined <- adults_plot %>%
   mutate(eco_cat = case_when(hab=="scansorial" & act=="diurnal" ~ "both",
                             hab == "scansorial" ~ "scansorial",
                             act == "diurnal" ~ "diurnal", 
                             hab == "other" & act == "nondiurnal" ~ "neither")) %>%
  select(tip, labels, t50_merge, pUVA_merge, eco_cat)

# set row names in dataset to match the tip labels in the tree
row.names(adults_plot.combined) <- adults_plot.combined$tip

#check that phylogeny and data match exactly
name.check(tree.adults, adults_plot.combined)

#resort trait dataset to the order of tree tip labels
adults_plot.combined <- adults_plot.combined[tree.adults$tip.label, ] 

#labels
labs <- adults_plot.combined %>% select(tip, labels)

#make color pallette for new combined states
col_ecocat <- c("both" = "#C107B6",
                "diurnal" = "#FFB122",
                "neither" = "#BDC3BE",
                "scansorial" = "#009E73")
```


```{r ggtree-fig, fig.width = 10, fig.height=15}

# Make the phylogeny plot
p <- ggtree(tree.adults) %<+% labs + 
     geom_tiplab(cex = 3, aes(label = labels), fontface = 3) + 
   xlim_tree(450) + 
   coord_cartesian(clip = 'off') 

# Make a second plot next to the phylogeny (dot plot of t50 values)
p2 <- facet_plot(p, panel="t50 (nm)", data=adults_plot.combined, geom=geom_point, aes(x=t50_merge, color = eco_cat), shape = 19, cex = 2.5) +
  scale_color_manual(values = col_ecocat,
                     name = "Ecology",
                     breaks = c("scansorial", "diurnal", "both", "neither"),
                     labels = c("scansorial", "diurnal", "both", "neither"))


# Make a third plot next to the first two (bar plot of %UVA)
p3 <- facet_plot(p2, panel='%UVA', data=adults_plot.combined, geom=geom_segment, aes(x=0, xend=pUVA_merge, y=y, yend=y, color = eco_cat), size=3) +
  theme_tree2(legend.position=c(.07, .93)) #add scale bars and move legends

#control widths of panels
 gt = ggplot_gtable(ggplot_build(p3))
# gtable_show_layout(gt) # will show you the layout - very handy function
# gt # see plot layout in table format
# gt$layout$l[grep('tree', gt$layout$name)] # you want to find the column specific to panel-2
 gt$widths[7] = 0.5*gt$widths[7] # reduce column 7
 gt$widths[9] = 0.5*gt$widths[9] 

 #print figure
p3
```

```{r, results = "hide"}
#export as pdf
pdf("../Figures/ggtree-fig.pdf", width = 10, height = 15)
grid::grid.draw(gt) # plot with grid draw
dev.off()
```


# Comparative evolutionary models 

Here, we used PGLS models in caper to run phylogenetic ANOVAs to test for associations between lens transmission and ecological variables while accounting for evolutionary non-independence among species. These models use all available data, including all species we sampled plus all additional species available in published literature. Because published measurements did not always record transmission down to 305nm, we could not calculate %UVA (400-305nm) for 10 species. Thus, our species sampling for t50 (n = 114) is higher than for %UVA (n = 104) in these models. 

## t50 and ecology 

```{r, results = "hide"}

# Prep data and phylogeny -----

#subset data for T50
adult_t50 <- adults_merged %>%
  mutate(tip = ASW_names) %>%
  select(tip, t50_merge, hab, act, dich)

# set row names in dataset to match the tip labels in the tree
row.names(adult_t50) <- adult_t50$tip

#check that phylogeny and data match exactly
name.check(tree.adults, adult_t50)

#resort trait dataset to the order of tree tip labels
adult_t50 <- adult_t50[tree.adults$tip.label, ] 


# Prep data for caper -------

#use caper function to combine phylogeny and data into one object 
#(this function also matches species names in tree and dataset)
t50.comp <- comparative.data(phy = tree.adults, data = adult_t50, 
                             names.col = tip, 
                             vcv = TRUE, 
                             na.omit = FALSE, 
                             warn.dropped = TRUE)

#check for dropped tips or dropped species
t50.comp$dropped$tips #phylogeny
t50.comp$dropped$unmatched.rows #dataset
```

### t50 across habitats

Species that climb up plants/trees (=scansorial) have significantly higher t50 values (more UV filtering) than non-scansorial species in a phylogenetic ANOVA.

```{r}
#habitat vs. T50
pgls_hab.t50 <- pgls(t50_merge ~ hab, 
               data = t50.comp,
               lambda = "ML", 
               param.CI = 0.95)

# #diagnostic plots
# par(mar = c(4,4,2,2))
# par(mfrow = c(2, 2))
# plot(pgls_hab.t50)
# par(mfrow = c(1, 1))
# 
# #quick plot of pgls
# plot(t50_merge ~ hab, data = adult_t50)
# abline(pgls_hab.t50)
# 
# #Likelihood plot for Pagel's lambda
# lambda.hab.t50 <- pgls.profile(pgls_hab.t50, "lambda")
# plot(lambda.hab.t50)

#main effects
anova(pgls_hab.t50)

#coefficients
summary(pgls_hab.t50)
```

### t50 across activity periods

Diurnal (day-active) species have significantly higher t50 values (more UV filtering) than non-diurnal species in a phylogenetic ANOVA.

```{r}
# Diurnality --------

#activity per vs. T50
pgls_act.t50 <- pgls(t50_merge ~ act, 
               data = t50.comp,
               lambda = "ML", 
               param.CI = 0.95)

#main effects
anova(pgls_act.t50)

#coefficients
summary(pgls_act.t50)

```

### t50 across sexual dichromatism

Sexually dichromatic (different coloration/patterning in males and females) species and nondichromatic species show no significant differences in t50 values (no difference in lens filtering).

```{r}
# Sexual dichromatism --------

#dichromatism vs. T50
pgls_dich.t50 <- pgls(t50_merge ~ dich, 
               data = t50.comp,
               lambda = "ML", 
               param.CI = 0.95)

#main effects
anova(pgls_dich.t50)

#coefficients
summary(pgls_dich.t50)
```

## %UVA and ecology 

Here, we used the same type of models to test for associations between ecology and %UVA transmitted through lenses across species. Note that the species dataset for these analyses are slightly reduced; 10 species with data gathered from the literature could not be included, as transmission measurements were not made down to 305nm.

```{r, results = "hide"}

# Prep data and phylogeny -----

#subset data for uva
adult_uva <- adults_merged %>%
  filter(!is.na(pUVA_merge)) %>%
  mutate(tip = ASW_names) %>%
  select(tip, pUVA_merge, hab, act, dich)

# set row names in dataset to match the tip labels in the tree
row.names(adult_uva) <- adult_uva$tip

#check that phylogeny and data match exactly
name.check(tree.adults, adult_uva)

#make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(tree.adults$tip.label, adult_uva$tip)

#drop unwanted tips from phylogeny
tree.adults.uva <- drop.tip(phy = tree.adults, tip = drops) 

#check that phylogeny tips and data match exactly (if they match will return "OK")
name.check(tree.adults.uva, adult_uva)

#resort trait dataset to the order of tree tip labels
adult_uva <- adult_uva[tree.adults.uva$tip.label, ] 

# Prep data for caper -------

#use caper function to combine phylogeny and data into one object 
#(this function also matches species names in tree and dataset)
uva.comp <- comparative.data(phy = tree.adults.uva, data = adult_uva, 
                             names.col = tip, 
                             vcv = TRUE, 
                             na.omit = FALSE, 
                             warn.dropped = TRUE)

#check for dropped tips or dropped species
uva.comp$dropped$tips #phylogeny
uva.comp$dropped$unmatched.rows #dataset
```

### %UVA across habitats

Scansorial (climbing) species have significantly lower %UVA (more UV filtering) than non-scansorial species in a phylogenetic ANOVA. This is consistent with the findings based on t50. 

```{r}
#habitat vs. UVA
pgls_hab.uva <- pgls(pUVA_merge ~ hab, 
               data = uva.comp,
               lambda = "ML", 
               param.CI = 0.95)

#main effects
anova(pgls_hab.uva)

#coefficients
summary(pgls_hab.uva)
```

### %UVA across activity periods

Diurnal (day-active) species have lower %UVA transmission than non-diurnal species (the trend we expect), but the difference is not significant. This is different from our findings using t50, which indicated a significant difference between diurnal and non-diurnal species. This likely is the result of reduced sampling for diurnal species using %UVA, as 4/13 diurnal species were omitted for %UVA analysis because transmission was not reported down to 305nm. 

```{r}
# Diurnality --------

#activity per vs. UVA
pgls_act.uva <- pgls(pUVA_merge ~ act, 
               data = uva.comp,
               lambda = "ML", 
               param.CI = 0.95)

#main effects
anova(pgls_act.uva)

#coefficients
summary(pgls_act.uva)

```

### %UVA across sexual dichromatism

Sexually dichromatic and nondichromatic species show no significant differences in lens filtering measured by %UVA (this matches findings with t50). 

```{r}
# Sexual dichromatism --------

#dichromatism vs. UVA
pgls_dich.uva <- pgls(pUVA_merge ~ dich, 
               data = uva.comp,
               lambda = "ML", 
               param.CI = 0.95)

#main effects
anova(pgls_dich.uva)

#coefficients
summary(pgls_dich.uva)

```

## Test for multiple effects (model selection)

Previously, we tested each variable separately for a significant effect on transmission. These tests indicated that diurnality and scansoriality both affect transmission in amphibians. Here, we use model ranking by AIC to determine the best-fit model explaining transmission in amphibians. 

The anova() function will take the model objects as arguments, and return an ANOVA testing whether the more complex model is significantly better at capturing the data than the simpler model. If the resulting p-value is sufficiently low (usually less than 0.05), we conclude that the more complex model is significantly better than the simpler model, and thus favor the more complex model. If the p-value is not sufficiently low (usually greater than 0.05), we should favor the simpler model.


### Models of t50

Here, lambda is set to 1. 
```{r}

# With lambda = 1 -----

#model including scansoriality
mod1 <- pgls(t50_merge ~ hab, 
               data = t50.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including diurnality
mod2 <- pgls(t50_merge ~ act, 
               data = t50.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including sexual dichromatism
mod3 <- pgls(t50_merge ~ dich, 
               data = t50.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including scansoriality + diurnality
mod4 <- pgls(t50_merge ~ hab + act, 
               data = t50.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including scansoriality + diurnality + sexual dichromatism
mod5 <- pgls(t50_merge ~ hab + act + dich, 
               data = t50.comp,
               lambda = 1, 
               param.CI = 0.95)

#compare models
anova(mod1, mod2, mod3, mod4, mod5)
AIC(mod1, mod2, mod3, mod4, mod5)
```

Here, lambda is set to 0. 
```{r}
#With lambda = 0 -----
#model including scansoriality
mod1 <- pgls(t50_merge ~ hab, 
               data = t50.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including diurnality
mod2 <- pgls(t50_merge ~ act, 
               data = t50.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including sexual dichromatism
mod3 <- pgls(t50_merge ~ dich, 
               data = t50.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including scansoriality + diurnality
mod4 <- pgls(t50_merge ~ hab + act, 
               data = t50.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including scansoriality + diurnality + sexual dichromatism
mod5 <- pgls(t50_merge ~ hab + act + dich, 
               data = t50.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#compare models
anova(mod1, mod2, mod3, mod4, mod5)
AIC(mod1, mod2, mod3, mod4, mod5)
```

From AIC model ranking, we find that a model including both scansoriality and diurnality (but excluding sexual dichromatism) is the best to our t50 data. This was consistend both with lambda set to 0 and lambda set to 1. 

### Models of %UVA

Here, lambda is set to 1. 

```{r}

# With lambda = 1 -----

#model including scansoriality
mod1 <- pgls(pUVA_merge ~ hab, 
               data = uva.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including diurnality
mod2 <- pgls(pUVA_merge ~ act, 
               data = uva.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including sexual dichromatism
mod3 <- pgls(pUVA_merge ~ dich, 
               data = uva.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including scansoriality + diurnality
mod4 <- pgls(pUVA_merge ~ hab + act, 
               data = uva.comp,
               lambda = 1, 
               param.CI = 0.95)

#model including scansoriality + diurnality + sexual dichromatism
mod5 <- pgls(pUVA_merge ~ hab + act + dich, 
               data = uva.comp,
               lambda = 1, 
               param.CI = 0.95)

#compare models
anova(mod1, mod2, mod3, mod4, mod5)
AIC(mod1, mod2, mod3, mod4, mod5)
```

Here, lambda is set to 0. 

```{r}
#With lambda = 0 -----
#model including scansoriality
mod1 <- pgls(pUVA_merge ~ hab, 
               data = uva.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including diurnality
mod2 <- pgls(pUVA_merge ~ act, 
               data = uva.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including sexual dichromatism
mod3 <- pgls(pUVA_merge ~ dich, 
               data = uva.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including scansoriality + diurnality
mod4 <- pgls(pUVA_merge ~ hab + act, 
               data = uva.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#model including scansoriality + diurnality + sexual dichromatism
mod5 <- pgls(pUVA_merge ~ hab + act + dich, 
               data = uva.comp,
               lambda = 0.00001, 
               param.CI = 0.95)

#compare models
anova(mod1, mod2, mod3, mod4, mod5)
AIC(mod1, mod2, mod3, mod4, mod5)
```

From AIC model ranking, we find that a model including both scansoriality and diurnality (but excluding sexual dichromatism) is the best to our %UVA data when lambda is set to 0. When lambda is set to 1, the lowest AIC is for a model including only scansoriality, though this is not significantly different from the model including just diurnality, the model including diurnality + scansoriality, or even the model including all three ecocological variables (probably due to low power). 

# Plot specimen data for select species

Here, we make plots of the full spectral transmission curve for 1) a larger and a smaller specimen of _Hylodes phyllodes_, and 2) an adult and a tadpole of _Alytes muletensis_. We will add the other tapole sampled ( _Dendropsophus minutus_ ) to the plot with _Alytes_, as that data isn't used elsewhere in the analyses. 

```{r}

#import species transmission curves
species_curves <- data.frame(read.csv("../Data/scan_summary.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

```


## Hylodes of different sizes

```{r}
#pull out Hylodes curves
hylodes <- species_curves %>%
  select(wavelength_nm, Hylodes.phyllodes.larger, Hylodes.phyllodes.smaller) %>%
  mutate(uva_larger = ifelse(wavelength_nm<401, Hylodes.phyllodes.larger, NA)) %>%
  mutate(uva_smaller = ifelse(wavelength_nm<401, Hylodes.phyllodes.smaller, NA))

#set colors
col_hylodes <- c("1mm lens" = "#989595",
                 "2mm lens" = "black")

#plot both specimens
plot_hylodes <- ggplot(hylodes, aes(x = wavelength_nm)) + 
  #transmission curves
  geom_path(aes(y = Hylodes.phyllodes.smaller, color = "1mm lens"), size = 1) + 
  geom_path(aes(y = Hylodes.phyllodes.larger, color = "2mm lens"), size = 1) +
  #fill in area under curves between 305-400nm to show %UVA
  geom_area(aes(y = uva_larger), fill = "black", alpha = 0.4) +
  geom_area(aes(y = uva_smaller), fill = "#989595", alpha = 0.4) +
  #plot T50 onto each curve
  geom_point(aes(filter(lens_hyl, stage=="adult")$t50, y=50), size = 3.5, col = "black", shape = 15) +
  geom_point(aes(filter(lens_hyl, stage=="adult_smaller")$t50, y=50), size = 3.5, col = "#989595", shape = 15) +
  scale_color_manual(values = col_hylodes) +
  labs(x = "Wavelength (nm)",
       y = "Normalized transmission (%)",
       color = "Hylodes phyllodes") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.title = element_text(face = "italic"))

plot_hylodes
```


## Alytes tadpole vs. adult

```{r}
#pull out Hylodes curves
alytes <- species_curves %>%
  select(wavelength_nm, Alytes.muletensis.adult, Alytes.muletensis.tadpole)

#set colors
col_alytes <- c("tadpole" = "#989595",
                 "adult" = "black")

#plot both specimens
plot_alytes <- ggplot(alytes, aes(x = wavelength_nm)) + 
  geom_path(aes(y = Alytes.muletensis.tadpole, color = "tadpole"), size = 1) +
  geom_path(aes(y = Alytes.muletensis.adult, color = "adult"), size = 1) +
  scale_color_manual(values = col_alytes) +
  labs(x = "Wavelength (nm)",
       y = "Normalized transmission",
       color = "Alytes muletensis") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.title = element_text(face = "italic"))

plot_alytes
```

## Dendropsophus tadpole vs. other adults

Here we don't have an exact species match, but we do have closely related species (same genus). We can use this, or we can add the Dendropsophus tadpole to the Alytes plot so that data is shown somewhere and can be mentioned briefly in discussion of tadpoles. 

```{r}
#pull out Hylodes curves
dendro <- species_curves %>%
  select(wavelength_nm, Dendropsophus.leucophyllatus, Dendropsophus.minusculus, Dendropsophus.minutus.tadpole)

#set colors
col_dendro <- c("D. leucophyllatus adult" = "red",
                "D. minusculus adult" = "black",
                "D. minutus tadpole" = "#989595")

#plot both specimens
plot_dendro <- ggplot(dendro, aes(x = wavelength_nm)) + 
  geom_path(aes(y = Dendropsophus.leucophyllatus, color = "D. leucophyllatus adult"), size = 1) +
  geom_path(aes(y = Dendropsophus.minusculus, color = "D. minusculus adult"), size = 1) +
  geom_path(aes(y = Dendropsophus.minutus.tadpole, color = "D. minutus tadpole"), size = 1) +
  scale_color_manual(values = col_dendro) +
  labs(x = "Wavelength (nm)",
       y = "Normalized transmission",
       color = "Species/stage") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank())

plot_dendro
```

