---
title: "Lens transmission across anuran ecology"
author: "Katie Thomas"
date: "10 DEC 2020"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(knitr)
library(kableExtra)
library(dplyr)
library(ape)
library(plotly)
library(plyr)
library(geiger)
library(phytools)
library(AmphiNom)
library(cowplot)

# Markdown settings
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, fig.height = 4.5, fig.width=8)
```

# Data

## Lens transmission data

### This study

Lens data collected from frozen and thawed lenses in this study. 

```{r lens-data}

#import species data for lens transmission
lens_raw <- data.frame(read.csv("../Data/species_data.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

#tidy data
lens_data <- lens_raw %>%
  #put genus and species in separate columns
  separate(species, c("genus", "species"), sep = " ", extra = "drop") %>%
  #add genus_species
  mutate(genus_species = as.factor(paste(genus, species, sep = "_"))) %>%
  mutate(dataset = "this_study") %>%
  mutate(source = "this_study") %>%
  mutate(scanned = "frozen lens") %>%
  select(order, family, genus_species, stage, dataset, source, diam_mm, t450nm, t50, pUVA)

```

### Compiled from literature

Lenses measured fresh in previous published literature. Note that for _Bombina orientalis_ the T50% was <300nm. To be able to include it here, I have set it at 300 so that it is numerical and can be included in models and visuals. 

```{r lens-data-lit}

#import species data for lens transmission
lens_lit_raw <- data.frame(read.csv("../Data/yovanovich_species.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

#tidy data
lens_lit <- lens_lit_raw %>%
  mutate(genus_species = as.factor(species)) %>%
  mutate(t50 = as.numeric(replace(t50, t50=="<300", 300))) %>%
  mutate(t450nm = t450nm*100) %>%
  mutate(source = "literature") %>%
  mutate(stage = "unknown") %>%
  mutate(diam_mm = "unknown") %>%
  select(order, family, genus_species, stage, dataset, source, diam_mm, t450nm, t50, pUVA)
```

### Merged data

Merge datasets and view sampling

```{r merge-data}

#merge new and published datasets 
lenses <-full_join(lens_data, lens_lit) 

#compile number of frogs sampled by species
samps <- lenses %>%
  arrange(genus_species) %>%
  arrange(family) %>%
  arrange(order) %>%
  select(order, family, genus_species, source, t50, pUVA)

#create table column names
names(samps) <- c("Order","Family", "Species", "Data source", "T50%", "%UVA") 

#generate scrolling table in RMarkdown
kable(samps[ , c("Order","Family", "Species", "Data source", "T50%", "%UVA")], 
      caption = "Species sampled and transmission summary data") %>%
  kable_styling(full_width = F) %>% 
  collapse_rows(columns = c(1,2), valign = "top") %>%
  scroll_box(height = "600px") 
```

## Ecological trait data

Ecological trait database imported here and matched to the species in the dataset. In the original trait database used for other projects, adult habitat is categorized as: fossorial, subfossorial, aquatic, semiaquatic, ground-dwelling, scansorial; here, we collapse it into binary states (scansorial or other) to test whether scansoriality is correlated with lens transmission. Likewise, in our original database, activity pattern was categorized as: diurnal, nocturnal, both; here, we make it binary (primarily diurnal or not) to test whether diurnal activity correlates with lens transmission. Finally, we include data for the presence or absence of sexual dichromatism in each species. 

```{r trait-data}
#import trait data
traits_raw <- data.frame(read.csv("../Data/lens_traits.csv", header=TRUE, na.strings=c("", "NA")))

#convert to binary traits and extract relevent columns
traits <- traits_raw %>%
  mutate(hab = recode_factor(Adult_habitat, "Scansorial" = "scansorial", "Fossorial" = "other", "Subfossorial" = "other", "Ground-dwelling" = "other", "Aquatic" = "other", "Semiaquatic" = "other")) %>%
  mutate(Activity_period = na_if(Activity_period, "Unknown")) %>%
  mutate(act = recode_factor(Activity_period, "Both" = "nondiurnal", "Diurnal" = "diurnal", "Nocturnal" = "nondiurnal")) %>%
  mutate(Sex_dichromatism = na_if(Sex_dichromatism, "Unknown")) %>%
  mutate (dich = recode_factor(Sex_dichromatism, "Absent" = "nondichromatic", "Present" = "dichromatic")) %>%
  select(genus_species, hab, act, dich) 

#merge lens and trait data by species name
lens_traits <- left_join(lenses, traits, by="genus_species") 
```


### Phylogeny

Here we import a full amphibian tree phylogenetic hypothesis by Jetz and Pyron (2019). It's important to note that this tree includes branches supported by molecular data as well as branches that are grafted on based exclusively on taxonomy. 

```{r pryontree, fig.height=50, fig.width=8}

#Import phylogeny from Jetz and Pyron 2019
tree_orig <- read.tree(file = "../Data/amph_shl_new_Consensus_7238.tre") #reads tree

#Test whether tree is rooted
is.rooted(tree_orig)
```

Next we must match up the species names in the tree to the species names in our dataset. Unfortunately, anuran taxonomy changes frequently and there are likely many species matches that are named differently in the tree and in our dataset. To match them up, we use the AmphiNom package by Liedtke (2019) to convert known synonyms to the taxonomy represented in Amphibian Speicies of the World (ASW). This will address the bulk of missing taxa; the remaining will need to be examined manually to match to the tree. 


```{r name-match, cache = TRUE, results="hide"}

#update AmphiNom package with most current ASW taxonomy
##note: this is time consuming but should be redone periodically for most updated results.
### last update: 10 December 2020
#getTaxonomy() 

### Dataset names ------

#Screen names in dataset to see how well they match up to ASW and what can be 'updated' seamlessly
datanames.asw <- aswSync(lens_traits$genus_species)
synonymReport(datanames.asw)

#pull out all direct matches or updated ASW taxonomy
data.update <- datanames.asw %>%
  filter(status == "up_to_date" | status =="updated") %>%
  mutate(genus_species = query) %>%
  select(genus_species, ASW_names)

#add column to lens data with updated ASW names
lens_names <- distinct(left_join(lens_traits, data.update, by = "genus_species"))

#check how many species aren't being matched to ASW names
missingASW <- lens_names %>%
  filter(is.na(ASW_names)) %>%
  select(genus_species, ASW_names)

#add ASW manual entries after checking species
missingASW$ASW_names[missingASW$genus_species=="Cornufer_guentheri"]<-"Cornufer_guentheri" #ambuiguous (mult. matches)
missingASW$ASW_names[missingASW$genus_species=="Lissotriton_vulgaris"]<-"Lissotriton_vulgaris" #ambuiguous (mult. matches)
missingASW$ASW_names[missingASW$genus_species=="Cynops_pyrrhogaster"]<-"Cynops_pyrrhogaster" #ambiguous (mult. matches)

#merge these updated names with original data update
data.update2 <- full_join(data.update, missingASW)

#remerge with pupil data
lens_names2 <- distinct(left_join(lens_traits, data.update2, by = "genus_species"))

### Phylogeny names -----

#Pull out tip labels from tree
tips <- as.vector(tree_orig$tip.label)

#Make a dataframe with tip vector as a column called "phylo_names"
phylo_names <- as.data.frame(tips, optional = TRUE, stringsAsFactors = FALSE)
colnames(phylo_names) <- "species_phylo"

#Screen names in phylogeny tips to see how well they match up to ASW and what can be 'updated' seamlessly
phylotips.asw <- aswSync(phylo_names$species_phylo)

#view summary
synonymReport(phylotips.asw)

#pull out all direct matches for updated ASW taxonomy
tip.update <- phylotips.asw %>%
  filter(status == "up_to_date" | status =="updated") %>%
  mutate(species_phylo = query) %>%
  select(species_phylo, ASW_names) 

#add column to phylotips with updated ASW names
phylo_names2 <- left_join(phylo_names, tip.update, by = "species_phylo")

### Find species missing from tips and dataset ----

#check how many species in ASW revised dataset are not matching new tree ASW labels
length(which(!lens_names2$ASW_names %in% phylo_names2$ASW_names))

#list species that need manually checked in tree
missing <- setdiff(lens_names2$ASW_names, phylo_names2$ASW_names)
missing <- as.data.frame(missing, optional = TRUE, stringsAsFactors = FALSE)
colnames(missing) <- "ASW_names" #rename col with ASW names
missing$species_phylo <- NA #create empty col for phylo matches

#add ASW phylo matches manually 

#Add Lissotriton vulgaris (was ambiguous match)
missing$species_phylo[missing$ASW_names=="Lissotriton_vulgaris"]<-"Lissotriton_vulgaris" 

#Add Cynops pyrrhogaster (was ambiguous match)
missing$species_phylo[missing$ASW_names=="Cynops_pyrrhogaster"]<-"Cynops_pyrrhogaster" 

#Add Lithobates pipiens
missing$species_phylo[missing$ASW_names=="Lithobates_pipiens"]<-"Rana_pipiens" 

#Boana diabolica is missing from phylogeny and will have to be grafted in as a sister taxon to Boana geographica (according to Caminer and Ron 2020)) later
# Adding it here as placeholder now. 
missing$species_phylo[missing$ASW_names=="Boana_diabolica"]<-"Boana_diabolica" #ambiguous (mult. matches)


#add manual matches to phylogeny ASW matches
phylo_names3 <- phylo_names2 %>%
  drop_na(ASW_names)
  
phylo_names4 <- full_join(phylo_names3, missing)

#check how many species in ASW revised dataset are not matching new tree ASW labels
missing2 <- setdiff(lens_names2$ASW_names, phylo_names4$ASW_names)

#check for duplicate taxa
n_occur <- data.frame(table(lens_names2$genus_species))
#View(n_occur)

### Match data species names with phylogeny tip labels ###

#remove duplicate ASW matches from phylogeny labels
phylo_names5 <- phylo_names4 %>%
  distinct(ASW_names, .keep_all = TRUE) #remove any duplicates

#use a left join in dplyr to match phylo tips to dataset species names
phylo_data <- lens_names2 %>%
  left_join(phylo_names5, by = "ASW_names") %>%
  drop_na(species_phylo) %>% #drop species not in phylogeny
  mutate_if(is.factor, as.character) %>%
  distinct(ASW_names, .keep_all = TRUE) #remove any duplicates

### Pruning a phylogeny to match the species in your dataset ###

#make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(tree_orig$tip.label, phylo_data$species_phylo)

#drop unwanted tips from phylogeny
tree.pruned <- drop.tip(phy = tree_orig, tip = drops) 

#see which tips you've kept in your phylogeny
tree.pruned$tip.label

#rename tips to ASW names
tree.pruned$tip.label <- phylo_data[["ASW_names"]][match(tree.pruned$tip.label, phylo_data[["species_phylo"]])]

#force species_phylo to dataframe
phylo_data <- as.data.frame(phylo_data)
rownames(phylo_data) <- phylo_data$ASW_names

#check that phylogeny tips and data match exactly (if they match will return "OK")
name.check(tree.pruned, phylo_data)

#check and reorder tree
is.rooted(tree.pruned) #tests whether tree is rooted (want it to return "TRUE") 

#Test whether tree is ultrametric
is.ultrametric(tree.pruned)

#Force tree to be ultrametric (it should be)
tree_edit <- force.ultrametric(tree.pruned)

#Boana diabolica is missing from phylogeny and will be grafted in as a polytomy based on the genus
tree_add <- add.species.to.genus(tree_edit, "Boana_diabolica", where = "root")

#test whether tree is dichotomous (no polytomies)
is.binary.tree(tree_add) 

#make tree dichotomous
tree_add2 <- multi2di(tree_add)

#ladderize 
lens.tree <- ladderize(tree_add2) #ladderizes tree
```


```{r plot-phylo, fig.height = 10, fig.width = 4}
#plot phylogeny
plot.phylo(lens.tree, #phylogeny to plot
           type = "phylogram", #how to visualize phylogeny
           show.tip.label = TRUE, #whether to show tip labels/species names
           cex = 1.0, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE,
           edge.width = 1.5, #thickness of phylogeny branches
           label.offset = 3) #how far away from tips to put labels
```

# Visualize data

## Analysis subsets

Here, we subset three groups for visualization and analyses by life stage and dataset: adults (includes our adult data and all published literature data), tadpoles (our data only), a pair of _Hyllodes phyllodes_, one juvenile and one adult that showed different strengths of UV filtering, likely due to the size of the frog/lens resulting in differences in path length of the light, and a tadpole/adult pair of _Alytes muletensis_ (our only tad/adult match in the dataset).

```{r subsets}

#make dataframe to match ASW names/tip labels to full lens dataframe
dat.names <- phylo_data %>%
  select(genus_species, ASW_names)

#match phylogeny tip labels (ASW names) with lens dataframe
lenses.phy <- left_join(lens_traits, dat.names, by = "genus_species")
  
#adults only
lens_adults <- lenses.phy %>%
  filter(stage %in% c("adult", "unknown"))

#adults-this study
lens_adults.new <- lens_adults %>% 
  filter(source == "this_study")

#adults-literature
lens_adults.pub <- lens_adults %>% 
  filter(source == "literature")

#tads only
lens_tads <- lenses.phy %>%
  filter(stage == "tadpole")

#hyllodes set
lens_hyl <- lenses.phy %>%
  filter(genus_species == "Hylodes_phyllodes")

#tad/adult match
lens_dev <- lenses.phy %>%
  filter(genus_species == "Alytes_muletensis")
```

## Plot of adult sampling onto phylogeny

```{r, fig.height=10, fig.width=5, fig.caption = "Phylogeny showing adult lens data (T50 wavelength) and ecological traits. The first (non italicized) column shows our T50 measurements for adults, the second (italicized) column shows T50 values calculated from data in published literature, green triangles show scansoriality, yellow circles show diurnality, and blue squares show sexual dichromatism."}

# prep data -----

#create dataframe of adults with this paper and lit data in two separate columns

#our data
new_cols <- lens_adults.new %>%
  mutate(t450nm_new = t450nm) %>%
  mutate(t50_new = t50) %>%
  mutate(pUVA_new = pUVA) %>%
  select(ASW_names, t450nm_new, t50_new, pUVA_new, hab, act, dich)

#published data
lit_cols <- lens_adults.pub %>%
  mutate(t450nm_lit = t450nm) %>%
  mutate(t50_lit = t50) %>%
  mutate(pUVA_lit = pUVA) %>%
  select(ASW_names, t450nm_lit, t50_lit, pUVA_lit, hab, act, dich)

#merge
adults_sep <- full_join(new_cols, lit_cols, by = c("ASW_names", "hab", "act", "dich"))

#prune tree to adult data -----

#make list of taxa to drop (in tree but not in dataset)
drops <- setdiff(lens.tree$tip.label, adults_sep$ASW_names)

#drop unwanted tips from phylogeny
tree.adults <- drop.tip(phy = lens.tree, tip = drops) 

#check that phylogeny tips and data match exactly (if they match will return "OK")
name.check(tree.adults, phylo_data)

#resort dataset to the order of tree tip labels
rownames(adults_sep) <- adults_sep$ASW_names
adults_sep <- adults_sep[tree.adults$tip.label, ] 


#plot phylogeny with sampling tip labels ----

#phylogeny
plot.phylo(tree.adults, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.7, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE, 
           edge.width = 2,
           label.offset = 80) 

#add tip labels for our data
tiplabels(adults_sep$t50_new,
          cex = 0.6,
          frame = "none",
          bg = "white",
          offset = 15) #size of labels

#add tip labels for published data
tiplabels(adults_sep$t50_lit,
          cex = 0.6,
          font = 3, 
          frame = "none",
          bg = "white",
          offset = 35) #size of labels

#add tip labels for scansorial/not
tiplabels(col = c("#009E73","grey95")[adults_sep$hab], #sets color habitat
          pch = 17,
          cex = 0.8,
          offset = 50) 

#add tip labels for diurnal/not
tiplabels(col = c("grey95","#F0E442")[adults_sep$act], #sets color habitat
          pch = 19,
          cex = 0.8,
          offset = 60) 

#add tip labels for sexually dimorphic/not
tiplabels(col = c("grey95","#56B4E9")[adults_sep$dich], #sets color habitat
          pch = 15,
          cex = 0.8,
          offset = 70) 

#export pdf

#pdf
pdf("../Figures/phylogeny_summary.pdf", width = 6, height = 11)

#phylogeny
plot.phylo(tree.adults, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.6, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE, 
           edge.width = 2,
           label.offset = 80) 

#add tip labels for our data
tiplabels(adults_sep$t50_new,
          cex = 0.6,
          frame = "none",
          bg = "white",
          offset = 15) #size of labels

#add tip labels for published data
tiplabels(adults_sep$t50_lit,
          cex = 0.6,
          font = 3, 
          frame = "none",
          bg = "white",
          offset = 35) #size of labels

#add tip labels for scansorial/not
tiplabels(col = c("#009E73","grey95")[adults_sep$hab], #sets color habitat
          pch = 17,
          cex = 0.8,
          offset = 50) 

#add tip labels for diurnal/not
tiplabels(col = c("grey95","#F0E442")[adults_sep$act], #sets color habitat
          pch = 19,
          cex = 0.8,
          offset = 60) 

#add tip labels for sexually dimorphic/not
tiplabels(col = c("grey95","#56B4E9")[adults_sep$dich], #sets color habitat
          pch = 15,
          cex = 0.8,
          offset = 70) 

dev.off()

```

### May try to re-build this plot in ggtree so I can color boxes by amount of filtering (closer to 500 = more yellow)

To save time, not attempting this now. 

```{r, fig.height=8, fig.width=4, include = FALSE}

## Try this in ggtree to get continuous fill of tiplabels

library(ggimage)
library(ggtree)


ggtree(lens.tree, ladderize = TRUE) +
  geom_tiplab()
```

## Plots of T50 and %UVA values across states

```{r boxplots-eco}

# Scansoriality ------

# boxplot of T50 across habs
boxplot_habs.t50 <- ggplot(data = lens_adults, 
                       aes(x = hab, y = t50)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Scansoriality") +
  ylab("T50 wavelength (nm)") +
  labs(fill = "Data source")

# boxplot of %UVA across habs
boxplot_habs.uva <- ggplot(data = lens_adults, 
                       aes(x = hab, y = pUVA)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Scansoriality") +
  ylab("UVA (%)") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_habs.t50, boxplot_habs.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids

# Diurnality ------

# boxplot of T50 across habs
boxplot_act.t50 <- ggplot(data = lens_adults, 
                       aes(x = act, y = t50)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Diurnality") +
  ylab("T50 wavelength (nm)") +
  labs(fill = "Data source")

# boxplot of %UVA across habs
boxplot_act.uva <- ggplot(data = lens_adults, 
                       aes(x = act, y = pUVA)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Diurnality") +
  ylab("UVA (%)") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_act.t50, boxplot_act.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids


# Sexual dichromatism ------

# boxplot of T50 across habs
boxplot_dich.t50 <- ggplot(data = lens_adults, 
                       aes(x = dich, y = t50)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Sexual dichromatism") +
  ylab("T50 wavelength (nm)") +
  labs(fill = "Data source")

# boxplot of %UVA across habs
boxplot_dich.uva <- ggplot(data = lens_adults, 
                       aes(x = dich, y = pUVA)) + 
  geom_boxplot(notch = FALSE, outlier.shape = NA, alpha = 0.6) + #controls boxes  
  stat_summary(fun.y = mean, colour = "black", geom = "point", shape = 18, size = 3, show_guide = FALSE) + #controls what stats shown
 geom_jitter(aes(color = source), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black")) + #controls background
  xlab("Sexual dichromatism") +
  ylab("UVA (%)") +
  labs(fill = "Data source")

#print plots
plot_grid(boxplot_dich.t50, boxplot_dich.uva,  #list of plots to arrange in grid
           align = 'vh', #align horizontally and vertically
           ncol = 2,
           rel_widths = c(1, 1)) #number of columns in grids

```


# Comparative evolutionary models

## Premilinary plots of data

Here I plot the data and traits onto the tree so we can look at the phylogenetic distribution and see if any interesting patterns come out. 

```{r phylo-lens, fig.height = 15, fig.width = 15, fig.cap = "Lens transmission and ecology across adult anurans"}

# create vector of colors for pupil constriction and shape

#create colors for habitat
col_hab <- c("Aquatic" = "#0072B2",
              "Fossorial" = "#D55E00",
              "Ground-dwelling" = "#E69F00",
              "Scansorial" = "#009E73",
              "Semiaquatic" = "#56B4E9",
              "Subfossorial" = "#CC79A7")

#create vector of colors for activity period
col_act <- c("Both" = "deeppink3",
             "Diurnal" = "darkgoldenrod1", 
             "Nocturnal" = "blueviolet")

# Prep data and phylogeny -----

#subset data for eye diameter and mass
adult_bars <- lens_adults %>%
  mutate(tip = genus_species, 
         UVA700 = pUV_700norm, 
         UVA450 = pUV_450norm, 
         hab = Adult_habitat,
         act = Activity_period) %>%
  select(tip, UVA700, UVA450, hab, act)

# set row names in dataset to match the tip labels in the tree
row.names(adult_bars) <- adult_bars$tip

#drop phylogeny tips not in dataset)
adult.tree <- drop.tip(lens.tree, lens.tree$tip.label[!(lens.tree$tip.label %in% adult_bars$tip)])

#check that phylogeny and data match exactly
name.check(adult.tree, adult_bars)

#resort trait dataset to the order of tree tip labels
adult_bars <- adult_bars[adult.tree$tip.label, ] 

#make trait vector for uva
uva450 <- as.vector(adult_bars$UVA450) 

#add tip label names to vector
names(uva450) <- adult_bars$tip 

#make trait vector for uva
uva700 <- as.vector(adult_bars$UVA700) 

#add tip label names to vector
names(uva700) <- adult_bars$tip 

#make trait vector for activity per
act <- as.vector(adult_bars$act) 

#add tip label names to vector
names(act) <- adult_bars$tip 

#make vector of colors corresponding to phylogeny tips
tipcols.act <- unname(col_act[act]) 

#make trait vector for habitat
hab <- as.vector(adult_bars$hab)

#add tip label names to vector
names(hab) <- adult_bars$tip

#make vector of colors corresponding to phylogeny tips
tipcols.hab <- unname(col_hab[hab]) 


# phylogeny of adult sampling --------

#export as pdf
pdf("../Figures/phylogeny_adults.pdf", width = 12, height = 12)

#plot your beautiful phylogeny
plot.phylo(adult.tree, #phylogeny to plot
           type = "fan", #how to visualize phylogeny
           show.tip.label = TRUE, #whether to show tip labels/species names
           cex = 1.0, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE,
           edge.width = 1.5, #thickness of phylogeny branches
           label.offset = 3) #how far away from tips to put labels

#end pdf
dev.off()


# phylogeny with %UVA normalized to 450 ------

#export as pdf
pdf("../Figures/phylogeny_UVA.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva450,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8)

#end pdf
dev.off()

# phylogeny with %UVA normalized to 450 with activity pattern ------

#export as pdf
pdf("../Figures/phylogeny_UVAact450.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva450,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.act)

#end pdf
dev.off()

# phylogeny with %UVA normalized to 700 with activity pattern ------

#export as pdf
pdf("../Figures/phylogeny_UVAact700.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva700,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.act)

#end pdf
dev.off()


# Test whether act associated with %UVA ------

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
uva.act.comp <- comparative.data(phy = adult.tree, data = lens_adults %>% droplevels(), names.col = genus_species, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
uva.act.comp$dropped$tips #phylogeny
uva.act.comp$dropped$unmatched.rows #dataset

#activity pattern vs. %UVA (normalized 450)
pgls_act <- pgls(pUV_450norm ~ Activity_period, 
               data = uva.act.comp,
               lambda = "ML", 
               param.CI = 0.95)
summary(pgls_act)
anova(pgls_act)

#activity pattern vs. %UVA (normalized 700)
pgls_act2 <- pgls(pUV_700norm ~ Activity_period, 
               data = uva.act.comp,
               lambda = "ML", 
               param.CI = 0.95)
summary(pgls_act2)
anova(pgls_act2)

# phylogeny with %UVA normalized to 450 with habitat ------

#export as pdf
pdf("../Figures/phylogeny_UVAhab450.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva450,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.hab)

#end pdf
dev.off()

# Test whether habitat associated with %UVA ------

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)
uva.hab.comp <- comparative.data(phy = adult.tree, data = lens_adults %>% droplevels(), names.col = genus_species, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
uva.hab.comp$dropped$tips #phylogeny
uva.hab.comp$dropped$unmatched.rows #dataset

#activity pattern vs. %UVA (normalized 450)
pgls_hab <- pgls(pUV_450norm ~ Adult_habitat, 
               data = uva.hab.comp,
               lambda = "ML", 
               param.CI = 0.95)

summary(pgls_hab)
anova(pgls_hab)

# Test whether habitat associated with %UVA ------

#use caper function to combine phylogeny and data into one object (this function also matches species names in tree and dataset)

#change habitat to scansorial vs. not
binary_hab <- lens_adults %>%
  mutate(binary_hab = recode_factor(Adult_habitat, "Scansorial" = "scansorial", "Fossorial" = "other", "Subfossorial" = "other", "Ground-dwelling" = "other", "Aquatic" = "other", "Semiaquatic" = "other"))

#resort trait dataset to the order of tree tip labels
rownames(binary_hab) <- binary_hab$genus_species
binary_hab <- binary_hab[adult.tree$tip.label, ] 

uva.habbi.comp <- comparative.data(phy = adult.tree, data = binary_hab %>% droplevels(), names.col = genus_species, vcv = TRUE, na.omit = FALSE, warn.dropped = TRUE)

#check for dropped tips or dropped species
uva.habbi.comp$dropped$tips #phylogeny
uva.habbi.comp$dropped$unmatched.rows #dataset

#activity pattern vs. %UVA (normalized 450)
pgls_habbi <- pgls(pUV_450norm ~ binary_hab, 
               data = uva.habbi.comp,
               lambda = "ML", 
               param.CI = 0.95)

summary(pgls_habbi)
anova(pgls_habbi)

#activity pattern vs. %UVA (normalized 700)
pgls_habbi700 <- pgls(pUV_700norm ~ binary_hab, 
               data = uva.habbi.comp,
               lambda = "ML", 
               param.CI = 0.95)

summary(pgls_habbi700)
anova(pgls_habbi700)

# phylogeny with %UVA normalized to 450 with binary habitats ------

#color vector for binary habitat
col_habbi <- c("scansorial" = "#009E73",
              "other" = "gray")


#make trait vector for habitat
habbi <- as.vector(binary_hab$binary_hab)

#add tip label names to vector
names(habbi) <- binary_hab$tip

#make vector of colors corresponding to phylogeny tips
tipcols.habbi <- unname(col_habbi[habbi]) 


#export as pdf
pdf("../Figures/phylogeny_UVAhabbi450.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva450,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.habbi)

#plot(NULL ,xaxt='n',yaxt='n',bty='n',ylab='',xlab='', xlim=0:1, ylim=0:1)
#add legend for pupil constriction
legend("topleft", legend = c("scansorial", "other"), 
       col = col_habbi,
       pch = 15, #shape of labels
       cex = 2, 
       box.lty = 0, 
       title = "Adult habitat", 
       title.adj = 0)

#end pdf
dev.off()

#export as pdf
pdf("../Figures/phylogeny_UVAhabbi700.pdf", width = 12, height = 12)

#plot phylogeny
plotTree.wBars(adult.tree, uva700,
               scale = 3, 
               tip.labels = TRUE, 
               type = "fan",
               offset = 2,
               ftype = "bi",
               fsize = 0.8,
               col = tipcols.habbi)

#end pdf
dev.off()

### testing other variables --------

#Sexual dichromatism
#activity pattern vs. %UVA (normalized 450)
pgls_sd <- pgls(pUV_450norm ~ Sex_di, 
               data = uva.hab.comp,
               lambda = "ML", 
               param.CI = 0.95)

summary(pgls_hab)
anova(pgls_hab)


```



