---
title: "Anuran lens transmission"
author: "Kate Thomas"
date: "28 Oct 2019"
output:
  html_document:
    toc: true
    toc_float: true
---

# Setup

```{r setup, include=FALSE}
library(tidyverse)
library(dplyr)
library(ape)
library(plotly)
library(plyr)
library(geiger)

# Markdown settings
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, fig.height = 4.5, fig.width=8)
```

# Data

## Lens transmission data

First, we imported lens transmission data for each species (species averages) measured in Ron's lab. Correction of these spectra (adjusting, averaging, smoothing) was done prior to collation into this dataset. 

Then we imported lens data for each specimen measured in the field with the Avantes spectrometer. These were calculated by taking lens transmission measurements relative to a standard with no lens prior to importation here. 

```{r data}

#import lab and field lens transmission data
lens.lab <- data.frame(read.csv("../Data/lens_data_lab.csv", header=TRUE, na.strings=c("", "NA"))) #load raw data

lens.field <- data.frame(read.csv("../Data/lens_data_field.csv", header=TRUE, na.strings=c("", "NA")))

#adjust field data so that 800nm = 100% transmission to match lab data

lens.field.adj <- lens.field

for (i in 2:46) {  #Loop over loop.vector
  
  org <- lens.field.adj[1596,i] # finds original value at 800nm
  adj <- 100/org #creates normalization factor
  
  lens.field.adj[i] <- lens.field.adj[i]*adj
  } #end loop


#format data in long format
lens.lab_long <- lens.lab %>%
  gather(Specimen, value, Xenopus_sp_X1_X2_X3_lab:Boana_faber_CFBH44036_lab) %>%
  mutate(Specimen = factor(Specimen))
  
lens.field_long <- lens.field.adj %>%
  gather(Specimen, value, Ceratobatrachus_guntheri_JWS881_field:Phyllomedusa_sp_tad_MW11127_field) %>%
  mutate(Specimen = factor(Specimen)) 

#combine lab and field data
lens.all <- full_join(lens.lab_long, lens.field_long)
lens.all$Specimen <- as.factor(lens.all$Specimen)

#import lookup table of data names, specimen numbers, and species
lookup <- data.frame(read.csv("../Data/lens_matching.csv", header=TRUE, na.strings=c("", "NA")))
lookup$Specimen <- as.factor(lookup$Specimen)

#add species names/tip labesl by specimen names
lens.all.names <- left_join(lens.all, lookup, by = "Specimen")

```

## Phylogeny

We imported a 7,238 amphibian phylogeny from Jetz and Pyron 2019 and then pruned the phylogeny to the species present in our dataset (n = 83).

All species are exact matches to the consensus phylogeny from Jetz and Pyron 2019 except for 4 that are close substitutions: Boana diabolica for Boana semilineatus (B. diabolica is in the B. semilineatus species group, see Fouquet et al. 2016); Megophrys sp. for Megophrys megacephala (may need to update to most likely species on Jeff recommendation); Dyscophus sp. for Dyscophus insularis (species ID likely coming with barcoding, may update); Sclerophrys sp. for Sclerophrys asmarae (species ID likely coming with barcoding, may update). 

```{r phylo, fig.height=10, fig.width=6}

#Import amphibian consensus phylogeny from Jetz and Pyron (2019) downloaded from Dryad
tree_orig <- read.tree(file = "../Data/amph_shl_new_Consensus_7238.tre") #reads tree

#Import species list and tip labels in dataset
sampling <- data.frame(read.csv("../Data/lens_species.csv", header=TRUE, na.strings=c("", "NA"))) 

#make changes to dataframe
sampling <- sampling %>%
  mutate(tiplabel_WalterPyron2019 = as.character(tiplabel_WalterPyron2019)) %>% #make tip labels characters to match phylogeny
  mutate(genus_species = as.character(paste(Genus, Species, sep="_"))) #add genus_species column

#make row names of data the phylo tip labels
rownames(sampling) <- sampling$tiplabel_WalterPyron2019

#make list of taxa to drop (not in dataset)
drops <- setdiff(tree_orig$tip.label, sampling$tiplabel_WalterPyron2019)

#drop extra tips from the tree
tree.pruned <- drop.tip(phy = tree_orig, tip = drops) 

#check that phylogeny tips and data match exactly
name.check(tree.pruned, sampling)

#rename and check tree
amph.tree <- tree.pruned
is.rooted(amph.tree) #tests whether tree is rooted 
is.binary.tree(amph.tree) #tests whether tree is dichotomous (no polytomies)
amph.tree <- ladderize(amph.tree) #ladderizes tree

#plot tree of sampled species with original tip labels
plot.phylo(amph.tree)

#replace Jetz and Pyron (2019) tip labels with our labels (from ASotW)
subs <- sampling[ , c("tiplabel_WalterPyron2019", "genus_species")] #create a dataframe with column 1 with all original tip labels, and column 2 with corresponding desired tip labels

#Make substitutions of species on tips based on our sampling
amph.tree$tip.label <- subs[[2]][match(amph.tree$tip.label, subs[[1]])]

#plot tree of sampled species with final tip labels
plot.phylo(amph.tree)

```

```{r msp-samp, fig.height=10, fig.width=6}

#Import species list and tip labels for MSP data
msp <- data.frame(read.csv("../Data/msp_species.csv", header=TRUE, na.strings=c("", "NA"))) 

#make changes to dataframe
msp <- msp %>%
  mutate(tiplabel_WalterPyron2019 = as.character(tiplabel_WalterPyron2019)) %>% #make tip labels characters to match phylogeny
  mutate(genus_species = as.character(paste(Genus, Species, sep="_"))) #add genus_species column

#make subset of adults/juvs
msp_ad <- msp %>% filter(Life_stage=="adult"|Life_stage=="juvenile")

#make subset of tadpoles
msp_tad <- msp %>% filter(Life_stage=="tadpole")

#make dataframe of MSP species
msp_species <- msp %>% distinct(tiplabel_WalterPyron2019, .keep_all = TRUE)

rownames(msp_species) <- msp_species$tiplabel_WalterPyron2019

#make list of taxa to drop (not in dataset)
drops <- setdiff(tree_orig$tip.label, msp_species$tiplabel_WalterPyron2019)

#drop extra tips from the tree
tree.pruned <- drop.tip(phy = tree_orig, tip = drops) 
#check that phylogeny tips and data match exactly
name.check(tree.pruned, msp_species)

#rename and check tree
msp.tree <- tree.pruned
is.rooted(msp.tree) #tests whether tree is rooted 
is.binary.tree(msp.tree) #tests whether tree is dichotomous (no polytomies)
msp.tree <- ladderize(msp.tree) #ladderizes tree

#plot tree of sampled species with original tip labels
plot.phylo(msp.tree)

#replace Jetz and Pyron (2019) tip labels with our labels (from ASotW)
subs <- msp_species[ , c("tiplabel_WalterPyron2019", "genus_species")] #create a dataframe with column 1 with all original tip labels, and column 2 with corresponding desired tip labels

#Make substitutions of species on tips based on our sampling
msp.tree$tip.label <- subs[[2]][match(msp.tree$tip.label, subs[[1]])]

#plot tree of sampled species with final tip labels
plot.phylo(msp.tree)


## add presence of adult and tadpole sampling to tips ##
ad_samp <- msp_ad %>%
  distinct(tiplabel_WalterPyron2019, .keep_all = TRUE) %>%
  mutate(ad_samp = "yes") %>%
  select(tiplabel_WalterPyron2019, ad_samp)

tad_samp <- msp_tad %>%
  distinct(tiplabel_WalterPyron2019, .keep_all = TRUE) %>%
  mutate(tad_samp = "yes") %>%
  #column_to_rownames("tiplabel_WalterPyron2019") %>%
  select(tiplabel_WalterPyron2019, tad_samp)

stage_samp <- merge(ad_samp, tad_samp, by = "tiplabel_WalterPyron2019", all = TRUE)

# add true/false for adult sampling to tip labels
msp_speciestips <- msp_species %>%
  mutate(tad_samp = ifelse(Life_stage=="tadpole", "yes", "no")) %>%
  mutate(ad_samp = ifelse(Life_stage=="adult", "yes", "no")) %>%
  mutate(juv_samp = ifelse(Life_stage=="sub-adult", "yes", "no"))
  
# create vector of colors for presence/absence of sample
cols <- c("yes" = "black", 
         "no" = "lightgray")



#plot tree of sampled species with final tip labels
pdf("msp_sampling_phylo.pdf", width = 7, height = 15)

plot.phylo(msp.tree, 
           type = "phylogram", 
           show.tip.label = TRUE, 
           cex = 0.7, #text size
           no.margin = TRUE, 
           use.edge.length = TRUE, 
           edge.width = 2,
           label.offset = 20) 

#add tip labels for adults
tiplabels(col = cols[ad_samp$ad_samp],  pch=15, cex = 1, adj = 3) #give the shape of tip labels and color

#add tip labels for juveniles/sub-adults
tiplabels(col = cols[msp_speciestips$juv_samp],  pch=15, cex = 1, adj = 9) #give the shape of tip labels and color

#add tip labels for tadpoles
tiplabels(col = cols[msp_speciestips$tad_samp],  pch=15, cex = 1, adj = 15) #give the shape of tip labels and color

dev.off()

```




## Traits

Finally we imported ecological trait data for the species in our lens dataset. This is a subset of the anuran trait database; I extracted just the species matching our lens dataset using the traits_tidy.R script in the Analysis folder. 

```{r traits}

#trait data
traits <- data.frame(read.csv("../Data/lens_traits.csv", header=TRUE, na.strings=c("", "NA")))

#merge trait data with species data
lens_traits <- merge(sampling, traits, by="genus_species", 
                  all.x = TRUE) 

##### START HERE TOMORROW: NEED TO DO A CLEANER MERGE SO I JUST ADD TRAITS AND NOT REPEAT TAXONOMIC INFORMATION #####
```


# Playing with the data

Then we plotted lab and field data (as many specimens as we had) by species. 

```{r plot-species, fig.height=40, fig.width=10}
#facet wrap plots in ggplot
species_plots <- ggplot(lens.all.names, aes(x = Wavelength_nm, y= value, col = Data), alpha = 0.1) +
  geom_point() +
  xlim(300,800) +
  ylim(0,100) +
  xlab("Wavelength (nm)") +
  ylab("Transmission (%)") +
  theme_bw() +
  facet_wrap(~Genus_species, ncol = 4)

#plot here
species_plots

#export pdf of figure
pdf("../Plots/species_plots.pdf", width=10, height=40)
species_plots
dev.off()
```

To see how the long-wavelength dropoff might affect the matching of field and lab observations, we also normalized both datasets to 500 nm and plotted them again. 

```{r normalize-500, fig.height=20, fig.width=5}

#adjust lab data so that 500nm = 100% transmission
lens.lab.fh <- lens.lab

for (i in 2:116) {  #Loop over loop.vector
  
  org <- lens.lab.fh[301,i] # finds original value at 800nm
  adj <- 100/org #creates normalization factor
  
  lens.lab.fh[i] <- lens.lab.fh[i]*adj
  } #end loop


#adjust field data so that 500nm = 100% transmission
lens.field.fh <- lens.field

for (i in 2:46) {  #Loop over loop.vector
  
  org <- lens.field.fh[503,i] # finds original value at 800nm
  adj <- 100/org #creates normalization factor
  
  lens.field.fh[i] <- lens.field.fh[i]*adj
  } #end loop

#format data in long format
lens.lab.fh_long <- lens.lab.fh %>%
  gather(Specimen, value, Xenopus_sp_X1_X2_X3_lab:Boana_faber_CFBH44036_lab) %>%
  mutate(Specimen = factor(Specimen))
  
lens.field.fh_long <- lens.field.fh %>%
  gather(Specimen, value, Ceratobatrachus_guntheri_JWS881_field:Phyllomedusa_sp_tad_MW11127_field) %>%
  mutate(Specimen = factor(Specimen)) 

#combine lab and field data
lens.all.fh <- full_join(lens.lab.fh_long, lens.field.fh_long)
lens.all.fh$Specimen <- as.factor(lens.all.fh$Specimen)

#add species names/tip labesl by specimen names
lens.all.fh.names <- left_join(lens.all.fh, lookup, by = "Specimen")

# plot species
species_fh <- ggplot(lens.all.fh.names, aes(x = Wavelength_nm, y= value, col = Data)) +
  geom_point() +
  xlim(300,500) +
  ylim(0,100) +
  xlab("Wavelength (nm)") +
  ylab("Transmission (%)") +
  theme_bw() +
  facet_wrap(~Genus_species, ncol = 4)

#plot here
species_fh

#export pdf of figure
pdf("../Plots/species_plots.fh.pdf", width=10, height=40)
species_fh
dev.off()
```

Plot normalized to 800 and normalized to 500 on the same plots. 

```{r normalize-both, fig.height=20, fig.width=5}

#facet wrap plots normalized to 800
species_compare <- ggplot(lens.all.names, aes(x = Wavelength_nm, y= value, col = Data), alpha = 0.1) +
  geom_point() +
  xlim(300,800) +
  ylim(0,100) +
  xlab("Wavelength (nm)") +
  ylab("Transmission (%)") +
  theme_bw() +
  geom_point(data = lens.all.fh.names, 
             aes(x = Wavelength_nm, y= value, col = Data), alpha = 0.1, col = "black") +
  facet_wrap(~Genus_species, ncol = 4)

#plot here
species_fh

#export pdf of figure
pdf("../Plots/species_compare.pdf", width=10, height=40)
species_compare
dev.off()
```


Then we imported lens summary data (with T50 and %UVA values) and ecological trait data, and merged them by species name. 

```{r traits, include=FALSE}

# lens summary data
lens.samp <- data.frame(read.csv("../Data/lens_metadata.csv", header=TRUE, na.strings=c("", "NA"))) 
lens.samp$genus_species <- as.factor(paste(lens.samp$Genus, 
                            lens.samp$species, 
                            sep="_"))

#trait data
traits <- data.frame(read.csv("../Data/trait_data.csv", header=TRUE, na.strings=c("", "NA")))
traits_subset <- traits %>%
  mutate(genus_species = as.factor(paste(traits$Genus, traits$Species, sep="_"))) %>%
  select(genus_species, Feng_tiplabel, Adult_habitat, Activity_period, Mating_habitat, Sex_dichromatism) 

#merge lens and trait data by species name
lens_traits <- merge(lens.samp, traits_subset, by="genus_species", 
                  all.x = TRUE) #dataframe where you can see original and new counts per species
```

We plotted how T50 and %UVA varied across ecological states (activity period and adult habitat). 

```{r summaries, fig.height = 5, fig.width = 8}

#make a colorblind-friendly pallette for habitat
col_hab <- c("Aquatic" = "#0072B2",
              "Fossorial" = "#D55E00",
              "Ground-dwelling" = "#E69F00",
              "Scansorial" = "#009E73",
              "Semiaquatic" = "#56B4E9",
              "Subfossorial" = "#CC79A7")

#make vector of colors for activity period
col_act <- c("Both" = "deeppink3",
             "Diurnal" = "darkgoldenrod1", 
             "Nocturnal" = "blueviolet",
             "Unknown" = "gray")

##### plot T50 by habitat #####
hab.T <- ggplot(data = filter(lens_traits, Adult_habitat != "Unknown" & !is.na(T50.)),aes(x = reorder(Adult_habitat, T50., FUN=mean), y = T50.)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_x_discrete(breaks=c("Aquatic","Semiaquatic","Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial"),labels=c("Aq.", "Semiaq.", "Foss.", "Subfoss.", "Ground", "Scans.")) +
  ylab("T50") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#plot here
ggplotly(hab.T)

#export plot
htmlwidgets::saveWidget(ggplotly(hab.T), "../Plots/T50_habitat.html")

#### plot T50 by activity period #####
act.T <- ggplot(data = lens_traits, aes(x = reorder(Activity_period, T50., FUN=mean), y = T50.)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("T50") +
  xlab("Activity period") +
  theme(legend.position = "none")

#plot here
ggplotly(act.T)

#export plot
htmlwidgets::saveWidget(ggplotly(act.T), "../Plots/T50_acper.html")

#### plot T50 by family ####
fam.T <- ggplot(data = lens_traits, aes(x = reorder(Family, T50., FUN=mean), y = T50.)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 90, size = 9)) +
  ylab("T50") +
  xlab("Family") +
  theme(legend.position = "none")

#plot here
ggplotly(fam.T)

#export plot
htmlwidgets::saveWidget(ggplotly(fam.T), "../Plots/T50_family.html")

#### plot %UVA by habitat ####
hab.uv <- ggplot(data = filter(lens_traits, Adult_habitat != "Unknown" & !is.na(X.UVA)),aes(x = reorder(Adult_habitat, X.UVA, FUN=mean), y = T50.)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Adult_habitat, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_hab) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  scale_x_discrete(breaks=c("Aquatic","Semiaquatic","Fossorial", "Subfossorial", "Ground-dwelling", "Scansorial"),labels=c("Aq.", "Semiaq.", "Foss.", "Subfoss.", "Ground", "Scans.")) +
  ylab("% UVA") +
  xlab("Adult habitat") +
  theme(legend.position = "none")

#plot here
ggplotly(hab.uv)

#export plot
htmlwidgets::saveWidget(ggplotly(hab.uv), "../Plots/UVA_habitat.html")

#### plot %UVA by activity period ####
act.uv <- ggplot(data = lens_traits, aes(x = reorder(Activity_period, X.UVA, FUN=mean), y = X.UVA)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(color = Activity_period, text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  scale_color_manual(values = col_act) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 0, size = 9)) +
  ylab("% UVA") +
  xlab("Activity period") +
  theme(legend.position = "none")

#plot here
ggplotly(act.uv)

#export plot
htmlwidgets::saveWidget(ggplotly(act.uv), "../Plots/UVA_acper.html")

#### plot %UVA by family ####
fam.T <- ggplot(data = lens_traits, aes(x = reorder(Family, X.UVA, FUN=mean), y = X.UVA)) +
  geom_boxplot(notch = FALSE, outlier.shape = NA) +
  stat_summary(fun.y = mean, colour="black", geom="point",  shape=18, size=3, show_guide = FALSE) +
  geom_jitter(aes(text = genus_species), shape = 19, size = 1.3, alpha = 0.5, position = position_jitter(0.15)) +
  theme(text = element_text(size=14), panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  theme(axis.text = element_text(angle = 90, size = 9)) +
  ylab("% UVA") +
  xlab("Family") +
  theme(legend.position = "none")

#plot here
ggplotly(fam.uv)

#export plot
htmlwidgets::saveWidget(ggplotly(fam.uv), "../Plots/UVA_family.html")
```


